From c595a1c92bf29f34f835fcd365ba7cf54a990700 Mon Sep 17 00:00:00 2001
From: John Ralls <jralls@ceridwen.us>
Date: Sun, 26 Dec 2010 15:02:36 -0800
Subject: [PATCH 02/15] Bug 628396: Gtk build fails because of objective-c
 elements

---
 gtk/Makefile.am |    7 ++-----
 1 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/gtk/Makefile.am b/gtk/Makefile.am
index 6050c76..78f4684 100644
--- a/gtk/Makefile.am
+++ b/gtk/Makefile.am
@@ -53,7 +53,6 @@ INCLUDES =						\
 	-DGTK_PRINT_BACKEND_ENABLE_UNSUPPORTED		\
 	$(GTK_DEBUG_FLAGS)				\
 	$(GTK_DEP_CFLAGS)				\
-	$(gtk_clipboard_dnd_c_sources_CFLAGS)		\
 	$(INCLUDED_IMMODULE_DEFINE)
 
 gtarget=$(gdktarget)
@@ -740,7 +739,7 @@ gtk_use_stub_c_sources = \
 	gtkplug-stub.c   \
 	gtksocket-stub.c \
 	gtkmountoperation-stub.c
-gtk_all_c_sources += $(gtk_use_x11_c_sources) $(gtk_use_win32_c_sources) $(gtk_use_quartz_c_sources) $(gtk_use_stub_c_sources)
+gtk_all_c_sources += $(gtk_use_x11_c_sources) $(gtk_use_win32_c_source) $(gtk_use_stub_c_sources)
 if USE_X11
 gtk_private_h_sources += gtkxembed.h gtktrayicon.h xembed.h
 gtk_c_sources += $(gtk_use_x11_c_sources)
@@ -750,9 +749,9 @@ gtk_private_h_sources += gtkwin32embed.h gtkwin32embedwidget.h
 gtk_c_sources += $(gtk_use_win32_c_sources)
 else
 if USE_QUARTZ
+libgtk_quartz_2_0_la_CFLAGS = "-xobjective-c"
 gtk_private_h_sources += gtksearchenginequartz.h
 gtk_c_sources += $(gtk_use_quartz_c_sources)
-gtk_use_quartz_c_sources_CFLAGS = "-xobjective-c"
 else
 gtk_c_sources += $(gtk_use_stub_c_sources)
 endif
@@ -762,10 +761,8 @@ endif
 if USE_QUARTZ
 gtk_clipboard_dnd_c_sources = gtkclipboard-quartz.c gtkdnd-quartz.c gtkquartz.c
 gtk_clipboard_dnd_h_sources = gtkquartz.h
-gtk_clipboard_dnd_c_sources_CFLAGS = "-xobjective-c"
 else
 gtk_clipboard_dnd_c_sources = gtkclipboard.c gtkdnd.c
-gtk_clipboard_dnd_c_sources_CFLAGS =
 endif
 EXTRA_DIST += gtkquartz.h
 
-- 
1.7.6.3.dirty

