--- a/configure	2009-10-04 17:31:16.000000000 -0700
+++ b/configure	2009-10-04 17:52:29.000000000 -0700
@@ -11221,6 +11221,8 @@
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char -mv8 -std=gnu99"
                 CFLAGS="-O20 -ffast-math -D__NO_MATH_INLINES -fsigned-char -mv8 -std=gnu99"
                 PROFILE="-pg -g -O20 -D__NO_MATH_INLINES -fsigned-char -mv8 -std=gnu99" ;;
+	*darwin*)
+		;;
         *)
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char -std=gnu99"
                 CFLAGS="-O20 -D__NO_MATH_INLINES -fsigned-char -std=gnu99"
