--- a/dbus/dbus-sysdeps-unix.c	2010-12-16 05:53:48.000000000 -0800
+++ b/dbus/dbus-sysdeps-unix.c	2011-02-05 15:15:00.000000000 -0800
@@ -204,7 +204,7 @@
                     int               start,
                     int               len)
 {
-#ifdef HAVE_DECL_MSG_NOSIGNAL
+#if HAVE_DECL_MSG_NOSIGNAL
   const char *data;
   int bytes_written;
 
@@ -442,7 +442,7 @@
  again:
 
   bytes_written = sendmsg (fd, &m, 0
-#ifdef HAVE_DECL_MSG_NOSIGNAL
+#if HAVE_DECL_MSG_NOSIGNAL
                            |MSG_NOSIGNAL
 #endif
                            );
@@ -481,7 +481,7 @@
                         int               start2,
                         int               len2)
 {
-#ifdef HAVE_DECL_MSG_NOSIGNAL
+#if HAVE_DECL_MSG_NOSIGNAL
   struct iovec vectors[2];
   const char *data1;
   const char *data2;
@@ -1494,13 +1494,13 @@
 
 #if defined(HAVE_CMSGCRED)
   bytes_written = sendmsg (server_fd, &msg, 0
-#ifdef HAVE_DECL_MSG_NOSIGNAL
+#if HAVE_DECL_MSG_NOSIGNAL
                            |MSG_NOSIGNAL
 #endif
                            );
 #else
   bytes_written = send (server_fd, buf, 1, 0
-#ifdef HAVE_DECL_MSG_NOSIGNAL
+#if HAVE_DECL_MSG_NOSIGNAL
                         |MSG_NOSIGNAL
 #endif
                         );
