diff -c /Users/john/Development/gtk-sources/enchant-1.6.0/src/enchant.c\~ /Users/john/Development/gtk-sources/enchant-1.6.0/src/enchant.c
--- a/src/enchant.c	Fri Mar 20 20:43:36 2015
+++ b/src/enchant.c	Sun Mar 22 16:39:44 2015
@@ -301,9 +301,9 @@
 	if (xdg_data_dir)
 		{
 			char **xdg_ddirs = g_strsplit (xdg_data_dir, ":", 0);
-			char *xdg_iter;
+			char **xdg_iter;
 
-			for (xdg_iter = xdg_ddirs; xdg_iter != NULL; xdg_iter++)
+			for (xdg_iter = xdg_ddirs; *xdg_iter != NULL; xdg_iter++)
 				{
 					conf_dirs = enchant_slist_append_unique_path (conf_dirs, g_strdup(*xdg_iter));
 				}

Diff finished.  Sun Mar 22 17:09:21 2015
