diff --git a/gdk/quartz/gdkcursor-quartz.c b/gdk/quartz/gdkcursor-quartz.c
index 2431eb8..360a977 100644
--- a/gdk/quartz/gdkcursor-quartz.c
+++ b/gdk/quartz/gdkcursor-quartz.c
@@ -45,6 +45,20 @@ gdk_quartz_cursor_new_from_nscursor (NSCursor      *nscursor,
   return cursor;
 }
 
+static GdkCursor *
+gdk_quartz_blank_cursor() 
+{
+  NSCursor *nscursor;
+  NSImage *nsimage;
+  NSSize size = {1.0, 1.0};
+
+  nsimage = [[NSImage alloc] initWithSize:size];
+  nscursor = [[NSCursor alloc] initWithImage:nsimage 
+		               hotSpot:NSMakePoint(1.0, 1.0)];
+  [nsimage release];
+  return gdk_quartz_cursor_new_from_nscursor(nscursor, (GdkCursorType)GDK_BLANK_CURSOR);
+}
+
 static gboolean
 get_bit (const guchar *data,
          gint          width,
@@ -79,7 +93,7 @@ create_builtin_cursor (GdkCursorType cursor_type)
   NSImage *image;
   NSCursor *nscursor;
 
-  if (cursor_type >= G_N_ELEMENTS (xcursors))
+  if (cursor_type >= G_N_ELEMENTS (xcursors) || cursor_type < 0)
     return NULL;
 
   cursor = cached_xcursors[cursor_type];
@@ -210,8 +224,12 @@ gdk_cursor_new_for_display (GdkDisplay    *display,
     case GDK_HAND2:
       nscursor = [NSCursor pointingHandCursor];
       break;
+    case GDK_CURSOR_IS_PIXMAP:
+      return NULL;
+    case GDK_BLANK_CURSOR:
+      return gdk_quartz_blank_cursor();
     default:
-      return gdk_cursor_ref (create_builtin_cursor (cursor_type));
+	return gdk_cursor_ref (create_builtin_cursor (cursor_type));
     }
 
   [nscursor retain];
