--- libgimp/gimpui.c	2018-08-19 09:47:18.000000000 -0700
+++ libgimp/gimpui.c	2018-08-21 04:41:38.000000000 -0700
@@ -80,6 +80,9 @@
                                                  gint           x,
                                                  gint           y,
                                                  PangoLayout   *layout);
+#ifdef GDK_WINDOWING_QUARTZ
+static void      osx_focus_window                (void);
+#endif


 static gboolean gimp_ui_initialized = FALSE;
@@ -182,7 +185,7 @@
   gimp_dialogs_show_help_button (gimp_show_help_button ());

 #ifdef GDK_WINDOWING_QUARTZ
-  [NSApp activateIgnoringOtherApps:YES];
+  g_idle_add ((GSourceFunc) osx_focus_window, NULL);
 #endif

   gimp_ui_fix_pixbuf_style ();
@@ -492,3 +495,11 @@
   if (area)
     gdk_gc_set_clip_rectangle (gc, NULL);
 }
+
+#ifdef GDK_WINDOWING_QUARTZ
+static void
+osx_focus_window (void)
+{
+  [NSApp activateIgnoringOtherApps:YES];
+}
+#endif
