diff -c /Users/john/Development/gtk-sources/gmp-4.3.2/gmp-h.in\~ /Users/john/Development/gtk-sources/gmp-4.3.2/gmp-h.in
--- a/gmp-h.in	Thu Jan  7 12:09:02 2010
+++ b/gmp-h.in	Sat Jul 18 15:15:10 2015
@@ -47,11 +47,7 @@
 #define __GNU_MP__ 4
 
 #define __need_size_t  /* tell gcc stddef.h we only want size_t */
-#if defined (__cplusplus)
-#include <cstddef>     /* for size_t */
-#else
 #include <stddef.h>    /* for size_t */
-#endif
 #undef __need_size_t
 
 /* Instantiated by configure. */

Diff finished.  Sat Jul 18 15:15:25 2015
