Index: src/GrampsDisplay.py
===================================================================
--- src/GrampsDisplay.py	(revision 15090)
+++ src/GrampsDisplay.py	(working copy)
@@ -85,7 +85,10 @@
     Open a file or url with the default application. This should work
     on GNOME, KDE, XFCE, ... as we use a freedesktop application
     """
-    prog = find_binary('xdg-open')
+    if constfunc.mac():
+        prog = find_binary('open')
+    else:
+        prog = find_binary('xdg-open')
     if prog:
         os.spawnvpe(os.P_NOWAIT, prog, [prog, file], os.environ)
         return True
