Index: src/gui/viewmanager.py
===================================================================
--- a/src/gui/viewmanager.py	(revision 15926)
+++ b/src/gui/viewmanager.py	(working copy)
@@ -50,7 +50,14 @@
 #
 #-------------------------------------------------------------------------
 import gtk
-
+if gtk.gdk.WINDOWING == "quartz":
+    try:
+        from gtk_osxapplication import *
+        _gtkosxapplication = True
+    except:
+        _gtkosxapplication = False
+else:
+    _gtkosxapplication = False
 #-------------------------------------------------------------------------
 #
 # GRAMPS modules
@@ -245,6 +246,9 @@
         the view categories are accessible in the sidebar.
         """
         CLIManager.__init__(self, dbstate, False)
+        if _gtkosxapplication:
+            self.macapp = OSXApplication()
+
         self.view_category_order = view_category_order
         #set pluginmanager to GUI one
         self._pmgr = GuiPluginManager.get_instance()
@@ -267,6 +271,8 @@
 
         self.__build_main_window()
         self.__connect_signals()
+        if _gtkosxapplication:
+            self.macapp.ready()
         
         self.do_reg_plugins()
         #plugins loaded now set relationship class
@@ -410,6 +416,8 @@
         """
         self.window.connect('delete-event', self.quit)
         self.notebook.connect('switch-page', self.change_category)
+        if _gtkosxapplication:
+            self.macapp.connect('NSApplicationWillTerminate', self.quit)
 
     def __init_lists(self):
         """
@@ -754,6 +762,15 @@
 
         self.uimanager.add_ui_from_string(UIDEFAULT)
         self.uimanager.ensure_update()
+        if _gtkosxapplication:
+            menubar = self.uimanager.get_widget("/MenuBar")
+            menubar.hide()
+            quit_item = self.uimanager.get_widget("/MenuBar/FileMenu/Quit")
+            about_item = self.uimanager.get_widget("/MenuBar/HelpMenu/About")
+            prefs_item = self.uimanager.get_widget("/MenuBar/EditMenu/Preferences")
+            self.macapp.set_menu_bar(menubar)
+            self.macapp.insert_app_menu_item(about_item, 0)
+            self.macapp.insert_app_menu_item(prefs_item, 1)
 
     def preferences_activate(self, obj):
         """
@@ -1192,6 +1209,8 @@
                     self.__connect_active_page(category_page, view_page)
 
                     self.uimanager.ensure_update()
+                    if _gtkosxapplication:
+                        self.macapp.sync_menubar()
 
                     while gtk.events_pending():
                         gtk.main_iteration()
