Index: src/GrampsDisplay.py
===================================================================
--- a/src/GrampsDisplay.py	(revision 15090)
+++ b/src/GrampsDisplay.py	(working copy)
@@ -23,6 +23,7 @@
 import const
 import locale
 import os
+from constfunc import is_quartz
 
 #list of manuals on wiki, map locale code to wiki extension, add language codes
 #completely, or first part, so pt_BR if Brazilian portugeze wiki manual, and 
@@ -85,7 +85,10 @@
     Open a file or url with the default application. This should work
     on GNOME, KDE, XFCE, ... as we use a freedesktop application
     """
-    prog = find_binary('xdg-open')
+    if is_quartz():
+        prog = find_binary('open')
+    else:
+        prog = find_binary('xdg-open')
     if prog:
         os.spawnvpe(os.P_NOWAIT, prog, [prog, file], os.environ)
         return True
