
--- a/gsf/gsf-output-iconv.c	Sun Mar  3 07:17:11 2013
+++ b/gsf/gsf-output-iconv.c	Sat Dec 13 10:32:43 2014
@@ -119,7 +119,7 @@
 			ok = FALSE;
 		} else {
 			ic->buf_len -= bytes_read;
-			g_memmove (ic->buf, ic->buf + bytes_read, ic->buf_len);
+			memmove (ic->buf, ic->buf + bytes_read, ic->buf_len);
 
 			ok = gsf_output_write (ic->sink, bytes_written, data);
 			if (!ok) {

Diff finished.  Sat Dec 20 15:21:40 2014
