diff -c /Users/john/Documents/Lion/gtk-stable-64/src/gtk\+-2.24.6/tests/Makefile.am\~ /Users/john/Documents/Lion/gtk-stable-64/src/gtk\+-2.24.6/tests/Makefile.am
--- a/tests/Makefile.am	Mon Aug 15 19:30:53 2011
+++ b/tests/Makefile.am	Fri Oct  7 12:20:00 2011
@@ -96,6 +96,7 @@
 
 autotestkeywords_SOURCES = autotestkeywords.cc
 autotestkeywords_CPPFLAGS = -I$(srcdir)/dummy-headers
+autotestkeywords_LDADD = $(LDADDS)
 
 if HAVE_OBJC
 autotestkeywords_CPPFLAGS += -DHAVE_OBJC=1 -x objective-c++

Diff finished.  Fri Oct  7 12:20:37 2011
