diff -c /Users/john/gtk/source/gtk\+-2.20.1/gtk/Makefile.am.orig /Users/john/gtk/source/gtk\+-2.20.1/gtk/Makefile.am
--- a/gtk/Makefile.am	Thu Sep  9 10:10:10 2010
+++ b/gtk/Makefile.am	Thu Sep  9 10:13:52 2010
@@ -57,7 +57,6 @@
 	-DGTK_PRINT_BACKEND_ENABLE_UNSUPPORTED		\
 	$(GTK_DEBUG_FLAGS)				\
 	$(GTK_DEP_CFLAGS)				\
-	$(gtk_clipboard_dnd_c_sources_CFLAGS)		\
 	$(INCLUDED_IMMODULE_DEFINE)
 
 gtarget=$(gdktarget)
@@ -732,6 +731,7 @@
 	gtkmountoperation-stub.c
 gtk_use_quartz_c_sources =	\
 	gtksearchenginequartz.c	\
+	gtkquartz.c		\
 	gtkplug-stub.c		\
 	gtksocket-stub.c	\
 	gtkmountoperation-stub.c
@@ -739,7 +739,7 @@
 	gtkplug-stub.c   \
 	gtksocket-stub.c \
 	gtkmountoperation-stub.c
-gtk_all_c_sources += $(gtk_use_x11_c_sources) $(gtk_use_win32_c_sources) $(gtk_use_quartz_c_sources) $(gtk_use_stub_c_sources)
+gtk_all_c_sources += $(gtk_use_x11_c_sources) $(gtk_use_win32_c_source) $(gtk_use_stub_c_sources)
 if USE_X11
 gtk_private_h_sources += gtkxembed.h gtktrayicon.h xembed.h
 gtk_c_sources += $(gtk_use_x11_c_sources)
@@ -749,9 +749,9 @@
 gtk_c_sources += $(gtk_use_win32_c_sources)
 else
 if USE_QUARTZ
+libgtk_quartz_3_0_la_CFLAGS = "-xobjective-c"
 gtk_private_h_sources += gtksearchenginequartz.h
 gtk_c_sources += $(gtk_use_quartz_c_sources)
-gtk_use_quartz_c_sources_CFLAGS = "-xobjective-c"
 else
 gtk_c_sources += $(gtk_use_stub_c_sources)
 endif
@@ -759,9 +759,8 @@
 endif
 
 if USE_QUARTZ
-gtk_clipboard_dnd_c_sources = gtkselection.c gtkselection-quartz.c gtkclipboard-quartz.c gtkdnd-quartz.c gtkquartz.c
+gtk_clipboard_dnd_c_sources = gtkselection.c gtkselection-quartz.c gtkclipboard-quartz.c gtkdnd-quartz.c
 gtk_clipboard_dnd_h_sources = gtkquartz.h
-gtk_clipboard_dnd_c_sources_CFLAGS = "-xobjective-c"
 else
 gtk_clipboard_dnd_c_sources = gtkselection.c gtkclipboard.c gtkdnd.c
 gtk_clipboard_dnd_c_sources_CFLAGS =

Diff finished.  Thu Sep  9 10:26:26 2010
