diff -c /Users/john/Development/gtk-sources/gtkspell3-3.0.0/gtkspell/Makefile.am\~ /Users/john/Development/gtk-sources/gtkspell3-3.0.0/gtkspell/Makefile.am
--- a/gtkspell/Makefile.am	Wed Oct 31 15:23:44 2012
+++ b/gtkspell/Makefile.am	Mon May 13 15:23:27 2013
@@ -6,24 +6,24 @@
 
 if ENABLE_GTK3
 lib_LTLIBRARIES += libgtkspell3-3.la
-libgtkspell3_3_la_CPPFLAGS = -DPACKAGE_LOCALE_DIR=\""$(localedir)"\" @GTK_SPELL3_CFLAGS@
+libgtkspell3_3_la_CPPFLAGS = -DPACKAGE_LOCALE_DIR=\""$(localedir)"\" $(GTK_SPELL3_CFLAGS)
 libgtkspell3_3_la_includedir=$(includedir)/gtkspell-3.0/gtkspell
 libgtkspell3_3_la_include_HEADERS = gtkspell.h
 
 libgtkspell3_3_la_SOURCES = gtkspell.c gtkspell.h
-libgtkspell3_3_la_LIBADD = @GTK_SPELL3_LIBS@
+libgtkspell3_3_la_LIBADD = $(GTK_SPELL3_LIBS)
 
 pkgconfig_DATA += gtkspell3-3.0.pc
 endif
 
 if ENABLE_GTK2
 lib_LTLIBRARIES += libgtkspell3-2.la
-libgtkspell3_2_la_CPPFLAGS = -DPACKAGE_LOCALE_DIR=\""$(localedir)"\" @GTK_SPELL2_CFLAGS@
+libgtkspell3_2_la_CPPFLAGS = -DPACKAGE_LOCALE_DIR=\""$(localedir)"\" $(GTK_SPELL2_CFLAGS)
 libgtkspell3_2_la_includedir=$(includedir)/gtkspell-3.0/gtkspell
 libgtkspell3_2_la_include_HEADERS = gtkspell.h
 
 libgtkspell3_2_la_SOURCES = gtkspell.c gtkspell.h
-libgtkspell3_2_la_LIBADD = @GTK_SPELL2_LIBS@
+libgtkspell3_2_la_LIBADD = $(GTK_SPELL2_LIBS)
 
 pkgconfig_DATA += gtkspell3-2.0.pc
 endif
@@ -38,7 +38,7 @@
 if ENABLE_GTK3
 GtkSpell-3.0.gir: libgtkspell3-3.la
 GtkSpell_3_0_gir_INCLUDES = GObject-2.0 GLib-2.0 Gtk-3.0
-GtkSpell_3_0_gir_CFLAGS = --warn-all @GTK_SPELL_CFLAGS@
+GtkSpell_3_0_gir_CFLAGS = --warn-all $(GTK_SPELL3_CFLAGS)
 GtkSpell_3_0_gir_LIBS = libgtkspell3-3.la
 GtkSpell_3_0_gir_FILES = $(libgtkspell3_3_la_SOURCES)
 INTROSPECTION_GIRS += GtkSpell-3.0.gir
@@ -47,7 +47,7 @@
 if ENABLE_GTK2
 GtkSpell-2.0.gir: libgtkspell3-2.la
 GtkSpell_2_0_gir_INCLUDES = GObject-2.0 GLib-2.0 Gtk-2.0
-GtkSpell_2_0_gir_CFLAGS = --warn-all @GTK_SPELL2_CFLAGS@
+GtkSpell_2_0_gir_CFLAGS = --warn-all $(GTK_SPELL2_CFLAGS)
 GtkSpell_2_0_gir_LIBS = libgtkspell3-2.la
 GtkSpell_2_0_gir_FILES = $(libgtkspell3_2_la_SOURCES)
 INTROSPECTION_GIRS += GtkSpell-2.0.gir
diff -Naur gtkspell3-3.0.0.orig/m4/.mkdir gtkspell3-3.0.0/m4/.mkdir
--- a/m4/.mkdir    1970-01-01 01:00:00.000000000 +0100
+++ b/m4/.mkdir    2013-07-16 22:17:47.000000000 +0100
@@ -0,0 +1 @@
+ensure-exists

Diff finished.  Mon May 13 15:23:33 2013
