diff --git a/src/ige-mac-menu.c b/src/ige-mac-menu.c
index bf44831..b9a6de8 100644
--- a/src/ige-mac-menu.c
+++ b/src/ige-mac-menu.c
@@ -997,7 +997,7 @@ parent_set_emission_hook_remove (GtkWidget *widget, gpointer data) {
 }
 
 static gboolean
-window_focus(GtkWindow *window, GtkWidget *widget, CarbonMenu *menu) {
+window_focus(GtkWindow *window, GdkEventFocus *event, CarbonMenu *menu) {
     OSStatus err = SetRootMenu(menu->menu);
     if (err) {
 	carbon_menu_warn(err, "Failed to transfer menu");
@@ -1050,7 +1050,7 @@ ige_mac_menu_set_menu_bar (GtkMenuShell *menu_shell) {
     g_printerr ("%s: syncing menubar\n", G_STRFUNC);
 #endif
     sync_menu_shell (menu_shell, carbon_menubar, TRUE, DEBUG_SET);
-    g_signal_connect (parent, "set-focus",
+    g_signal_connect (parent, "focus-in-event",
 		      G_CALLBACK(window_focus), 
 		      carbon_menu_get(GTK_WIDGET(menu_shell)));
 
