diff -c /Users/john/Development/Gramps-Build/gramps-41-tarball-Leopard/src/pygobject-3.14.0/gi/pygi-cache.h\~ /Users/john/Development/Gramps-Build/gramps-41-tarball-Leopard/src/pygobject-3.14.0/gi/pygi-cache.h
--- a/gi/pygi-cache.h	Wed Sep 10 15:10:32 2014
+++ b/gi/pygi-cache.h	Tue Feb 17 16:04:45 2015
@@ -197,7 +197,7 @@
                                      GICallableInfo *callable_info);
 };
 
-typedef struct _PyGIFunctionCache {
+struct _PyGIFunctionCache {
     PyGICallableCache callable_cache;
 
     /* An invoker with ffi_cif already setup */
@@ -207,13 +207,13 @@
                          PyGIInvokeState *state,
                          PyObject *py_args,
                          PyObject *py_kwargs);
-} PyGIFunctionCache;
+};
 
-typedef struct _PyGIVFuncCache {
+struct _PyGIVFuncCache {
     PyGIFunctionWithInstanceCache fwi_cache;
 
     GIBaseInfo *info;
-} PyGIVFuncCache;
+};
 
 
 gboolean

Diff finished.  Sun Feb 22 14:44:00 2015
