--- a/gtk/gtk.override	2010-07-15 09:51:18.000000000 +0200
+++ b/gtk/gtk.override	2010-07-15 10:02:45.000000000 +0200
@@ -1194,8 +1194,8 @@
     PySignalWatchSource *real_source = (PySignalWatchSource*)source;
 
     if (source != NULL) {
-	if (real_source->fds[0] != 0)
-	    close(real_source->fds[0]);
+//	if (real_source->fds[0] != 0)
+//	    close(real_source->fds[0]);
 	
 	if (real_source->fds[1] != 0) {
 #if HAVE_PYSIGNAL_SETWAKEUPFD
