diff -c /Users/john/Development/gtk-sources/Python-3.4.3/configure.ac\~ /Users/john/Development/gtk-sources/Python-3.4.3/configure.ac
*** a/configure.ac	Wed Feb 25 20:27:46 2015
--- b/configure.ac	Sun Mar 22 12:48:43 2015
***************
*** 1302,1307 ****
--- 1302,1313 ----
  
              CFLAGS="${UNIVERSAL_ARCH_FLAGS} -isysroot ${UNIVERSALSDK} ${CFLAGS}"
              LDFLAGS="${UNIVERSAL_ARCH_FLAGS} -isysroot ${UNIVERSALSDK} ${LDFLAGS}"
+ 		# Issue #18075: the default maximum stack size (8MBytes) is too
+ 		# small for the default recursion limit. Increase the stack size
+ 		# to ensure that tests don't crash
+ 		LDFLAGS="-Wl,-stack_size,1000000 $(LDFLAGS)
+ 
+ 
              if test "${UNIVERSALSDK}" != "/"
              then
                  CPPFLAGS="-isysroot ${UNIVERSALSDK} ${CPPFLAGS}"
***************
*** 2160,2170 ****
  	Darwin/*) 
  		LINKFORSHARED="$extra_undefs -framework CoreFoundation"
  
- 		# Issue #18075: the default maximum stack size (8MBytes) is too
- 		# small for the default recursion limit. Increase the stack size
- 		# to ensure that tests don't crash
- 		LINKFORSHARED="-Wl,-stack_size,1000000 $LINKFORSHARED"
- 
  		if test "$enable_framework"
  		then
  			LINKFORSHARED="$LINKFORSHARED "'$(PYTHONFRAMEWORKDIR)/Versions/$(VERSION)/$(PYTHONFRAMEWORK)'
--- 2166,2171 ----

Diff finished.  Sun Mar 22 17:34:44 2015
