--- a/Source/WTF/wtf/InlineASM.h	2012-11-23 13:12:16.000000000 -0700
+++ b/Source/WTF/wtf/InlineASM.h	2014-12-22 21:56:32.000000000 -0700
@@ -77,7 +77,7 @@
 // FIXME: figure out how this works on all the platforms. I know that
 // on Linux, the preferred form is ".Lstuff" as opposed to "Lstuff".
 // Don't know about any of the others.
-#if PLATFORM(MAC)
+#if PLATFORM(MAC) || OS(DARWIN)
 #define LOCAL_LABEL_STRING(name) "L" #name
 #elif OS(LINUX)
 #define LOCAL_LABEL_STRING(name) ".L" #name
--- webkitgtk-1.10.2-orig/Source/WTF/wtf/ThreadingPthreads.cpp	2012-11-23 13:12:16.000000000 -0700
+++ webkitgtk-1.10.2/Source/WTF/wtf/ThreadingPthreads.cpp	2014-12-22 23:36:27.000000000 -0700
@@ -55,7 +55,7 @@
 #include <sys/time.h>
 #endif
 
-#if OS(MAC_OS_X) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
+#if PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
 #include <objc/objc-auto.h>
 #endif
 
@@ -205,7 +205,7 @@
     UNUSED_PARAM(threadName);
 #endif
 
-#if OS(MAC_OS_X) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
+#if PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
     // All threads that potentially use APIs above the BSD layer must be registered with the Objective-C
     // garbage collector in case API implementations use garbage-collected memory.
     objc_registerThreadWithCollector();
