--- a/Source/cmake/FindGTK3.cmake   
+++ b/Source/cmake/FindGTK3.cmake   
@@ -5,6 +5,7 @@ 
 #  GTK3_INCLUDE_DIRS - the GTK+ 3. include directories
 #  GTK3_LIBRARIES - link these to use GTK+ 3.
 #  GTK3_SUPPORTS_GESTURES - GTK+ supports gestures (GTK+ >= 3.14)
+#  GTK3_SUPPORTS_QUARTZ - GTK+ supports Quartz backend
 #  GTK3_SUPPORTS_X11 - GTK+ supports X11 backend
 #  GTK3_SUPPORTS_WAYLAND - GTK+ supports Wayland backend
 #
@@ -49,6 +50,11 @@ if (GTK3_VERSION)
 endif ()
 
 if (GTK3_VERSION AND VERSION_OK)
+    pkg_check_modules(GTK3_QUARTZ gtk+-quartz-3.0)
+    if ("${GTK3_QUARTZ_VERSION}" VERSION_EQUAL "${GTK3_VERSION}")
+        set(GTK3_SUPPORTS_QUARTZ TRUE)
+    endif ()
+
     pkg_check_modules(GTK3_X11 gtk+-x11-3.0)
     if ("${GTK3_X11_VERSION}" VERSION_EQUAL "${GTK3_VERSION}")
         set(GTK3_SUPPORTS_X11 TRUE)
--- a/Source/cmake/OptionsGTK.cmake 
+++ b/Source/cmake/OptionsGTK.cmake 
@@ -183,6 +183,7 @@ WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_THREADED_COMPOSITOR PRIVATE OFF)
 # this point, and do not attempt to change any option after this point.
 WEBKIT_OPTION_END()
 
+SET_AND_EXPOSE_TO_BUILD(WTF_PLATFORM_QUARTZ ${ENABLE_QUARTZ_TARGET})
 SET_AND_EXPOSE_TO_BUILD(WTF_PLATFORM_X11 ${ENABLE_X11_TARGET})
 SET_AND_EXPOSE_TO_BUILD(WTF_PLATFORM_WAYLAND ${ENABLE_WAYLAND_TARGET})
 
@@ -389,6 +390,12 @@ if (ENABLE_VIDEO OR ENABLE_WEB_AUDIO)
     SET_AND_EXPOSE_TO_BUILD(USE_GSTREAMER TRUE)
 endif ()
 
+if (ENABLE_QUARTZ_TARGET)
+    if (NOT GTK3_SUPPORTS_QUARTZ)
+        message(FATAL_ERROR "Recompile GTK+ with Quartz backend to use ENABLE_QUARTZ_TARGET")
+    endif ()
+endif ()
+
 if (ENABLE_X11_TARGET)
     if (NOT GTK3_SUPPORTS_X11)
         message(FATAL_ERROR "Recompile GTK+ with X11 backend to use ENABLE_X11_TARGET")
