--- a/Source/WebCore/PlatformGTK.cmake  
+++ b/Source/WebCore/PlatformGTK.cmake  
@@ -414,12 +414,17 @@ if (USE_TEXTURE_MAPPER)
     )
     list(APPEND WebCore_SOURCES
         platform/graphics/texmap/BitmapTexture.cpp
-        platform/graphics/texmap/BitmapTextureGL.cpp
         platform/graphics/texmap/BitmapTexturePool.cpp
         platform/graphics/texmap/GraphicsLayerTextureMapper.cpp
-        platform/graphics/texmap/TextureMapperGL.cpp
-        platform/graphics/texmap/TextureMapperShaderProgram.cpp
     )
+
+    if (USE_TEXTURE_MAPPER_GL)
+        list(APPEND WebCore_SOURCES
+            platform/graphics/texmap/BitmapTextureGL.cpp
+            platform/graphics/texmap/TextureMapperGL.cpp
+            platform/graphics/texmap/TextureMapperShaderProgram.cpp
+        )
+    endif ()
 endif ()
 
 if (ENABLE_THREADED_COMPOSITOR)
