--- ./configure.ac~	2012-11-01 13:21:40.000000000 -0700
+++ ./configure.ac	2012-11-01 15:10:37.000000000 -0700
@@ -321,7 +321,7 @@
    AC_DEFINE([WTF_PLATFORM_DIRECTFB],[1],[Define if target is DirectFB])
 else
    PKG_CHECK_MODULES(CAIRO, cairo >= $CAIRO_REQUIRED_VERSION)
-   PKG_CHECK_MODULES(GTK, gtk+-$GTK_API_VERSION >= $GTK_REQUIRED_VERSION)
+   PKG_CHECK_MODULES(GTK, gtk+-$GTK_API_VERSION >= $GTK_REQUIRED_VERSION, gdk-pixbuf-2.0)
 
    if test "$with_target" = "x11" && test "$os_win32" = "no"; then
       # check for XT
--- ./Source/JavaScriptCore/GNUmakefile.am~	2011-09-26 13:54:57.000000000 -0700
+++ ./Source/JavaScriptCore/GNUmakefile.am	2012-11-02 13:23:15.000000000 -0700
@@ -72,6 +72,7 @@
 javascriptcore_cflags += \
 	$(global_cflags) \
 	$(GLIB_CFLAGS) \
+	$(GTK_CFLAGS) \
 	$(UNICODE_CFLAGS)
 
 pkgconfig_DATA += Source/JavaScriptCore/javascriptcoregtk-@WEBKITGTK_API_VERSION@.pc
@@ -105,6 +106,7 @@
 	-ansi \
 	-fno-strict-aliasing \
 	$(global_cflags) \
+	$(javascriptcore_cflags) \
 	$(GLOBALDEPS_CFLAGS)
 
 Programs_minidom_LDADD = \
@@ -132,6 +134,7 @@
 	-fno-strict-aliasing \
 	$(global_cxxflags) \
 	$(global_cflags) \
+	$(javascriptcore_cflags)\
 	$(GLOBALDEPS_CFLAGS) \
 	$(UNICODE_CFLAGS)
 
--- ./Source/JavaScriptCore/wtf/gtk/ThreadingGtk.cpp~	2011-09-26 13:54:57.000000000 -0700
+++ ./Source/JavaScriptCore/wtf/gtk/ThreadingGtk.cpp	2012-11-02 13:50:33.000000000 -0700
@@ -58,8 +58,8 @@
 
 void initializeThreading()
 {
-    if (!g_thread_supported())
-        g_thread_init(NULL);
+/*    if (!g_thread_supported())
+        g_thread_init(NULL); Not required after glib-2.32 */
     ASSERT(g_thread_supported());
 
     if (!atomicallyInitializedStaticMutex) {
--- ./Source/JavaScriptCore/wtf/Platform.h~	2011-09-26 13:54:57.000000000 -0700
+++ ./Source/JavaScriptCore/wtf/Platform.h	2012-11-01 14:17:38.000000000 -0700
@@ -1120,7 +1120,10 @@
 #endif
 
 #if ENABLE(GLIB_SUPPORT)
-#include "GTypedefs.h"
+#include <glib.h>
+#include <gtk-2.0/gtk/gtk.h>
+#include <gtk-2.0/gdk/gdk.h>
+#include <gdk-pixbuf-2.0/gdk-pixbuf/gdk-pixbuf.h>
 #endif
 
 /* FIXME: This define won't be needed once #27551 is fully landed. However, 
--- ./Source/WebCore/bridge/npruntime_internal.h~	2011-09-26 13:54:59.000000000 -0700
+++ ./Source/WebCore/bridge/npruntime_internal.h	2012-11-01 14:58:09.000000000 -0700
@@ -29,7 +29,7 @@
 #include "npfunctions.h"
 #include "npruntime.h"
 
-#ifdef XP_UNIX
+#ifdef XP_X11
     #include <X11/Xresource.h>
 
     #undef None
--- ./Source/WebCore/GNUmakefile.am~	2012-11-01 13:21:40.000000000 -0700
+++ ./Source/WebCore/GNUmakefile.am	2012-11-01 16:25:42.000000000 -0700
@@ -771,6 +771,7 @@
 	$(ENCHANT_CFLAGS) \
 	$(GEOCLUE_CFLAGS) \
 	$(GLIB_CFLAGS) \
+	$(GTK_CFLAGS) \
 	$(GSTREAMER_CFLAGS) \
 	$(LIBSOUP_CFLAGS) \
 	$(LIBXML_CFLAGS) \
--- ./Source/WebCore/plugins/gtk/PluginPackageGtk.cpp~	2011-09-26 13:55:00.000000000 -0700
+++ ./Source/WebCore/plugins/gtk/PluginPackageGtk.cpp	2012-11-02 12:42:03.000000000 -0700
@@ -103,7 +103,7 @@
 #endif
 }
 
-#if defined(XP_UNIX)
+#if defined(XP_X11)
 static int webkitgtkXError(Display* xdisplay, XErrorEvent* error)
 {
     gchar errorMessage[64];
@@ -152,7 +152,7 @@
 
     m_isLoaded = true;
 
-#if defined(XP_UNIX)
+#if defined(XP_X11)
     if (!g_strcmp0(baseName.get(), "libflashplayer.so")) {
         // Flash plugin can produce X errors that are handled by the GDK X error handler, which
         // exits the process. Since we don't want to crash due to flash bugs, we install a
--- ./Source/WebCore/plugins/gtk/PluginViewGtk.cpp~	2012-11-01 13:21:40.000000000 -0700
+++ ./Source/WebCore/plugins/gtk/PluginViewGtk.cpp	2012-11-02 12:45:36.000000000 -0700
@@ -691,7 +691,7 @@
 #endif
 
         case NPNVnetscapeWindow: {
-            GdkWindow* gdkWindow = gtk_widget_get_window(m_parentFrame->view()->hostWindow()->platformPageClient());
+            GdkWindow* G_GNUC_UNUSED gdkWindow = gtk_widget_get_window(m_parentFrame->view()->hostWindow()->platformPageClient());
 #if defined(XP_UNIX)
             *static_cast<Window*>(value) = GDK_WINDOW_XWINDOW(gdk_window_get_toplevel(gdkWindow));
 #elif defined(GDK_WINDOWING_WIN32)
@@ -741,7 +741,7 @@
         gtk_widget_queue_draw(m_parentFrame->view()->hostWindow()->platformPageClient());
 }
 
-#ifndef GDK_WINDOWING_WIN32
+#if !defined GDK_WINDOWING_WIN32 && defined XP_X11
 static Display* getPluginDisplay()
 {
     // The plugin toolkit might have a different X connection open.  Since we're
--- ./Source/WebCore/plugins/npapi.h~	2011-09-26 13:55:00.000000000 -0700
+++ ./Source/WebCore/plugins/npapi.h	2012-11-01 15:00:25.000000000 -0700
@@ -89,7 +89,7 @@
 #endif
 #endif
 
-#if defined(XP_UNIX)
+#if defined(XP_X11)
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 #include <stdio.h>
@@ -248,7 +248,7 @@
 /* Exact meaning must be spec'd in event model. */
 #define kNPEventStartIME 2
 
-#if defined(XP_UNIX)
+#if defined(XP_X11)
 /*
  * Unix specific structures and definitions
  */
@@ -541,7 +541,7 @@
   uint32_t wParam;
   uint32_t lParam;
 } NPEvent;
-#elif defined(XP_UNIX)
+#elif defined(XP_X11)
 typedef XEvent NPEvent;
 #else
 typedef void*  NPEvent;
@@ -555,7 +555,7 @@
 typedef CGPathRef NPCGRegion;
 #elif defined(XP_WIN)
 typedef HRGN NPRegion;
-#elif defined(XP_UNIX)
+#elif defined(XP_X11)
 typedef Region NPRegion;
 #elif defined(XP_SYMBIAN)
 typedef QRegion* NPRegion;
--- ./Source/WebCore/plugins/PluginView.cpp~	2011-09-26 13:55:00.000000000 -0700
+++ ./Source/WebCore/plugins/PluginView.cpp	2012-11-02 11:35:33.000000000 -0700
@@ -363,7 +363,7 @@
         PluginView::setCurrentPluginView(0);
     }
 
-#ifdef XP_UNIX
+#ifdef XP_X11
     if (m_isWindowed && m_npWindow.ws_info)
            delete (NPSetWindowCallbackStruct *)m_npWindow.ws_info;
     m_npWindow.ws_info = 0;
@@ -861,7 +861,7 @@
     , m_contextRef(0)
     , m_fakeWindow(0)
 #endif
-#if defined(XP_UNIX) && ENABLE(NETSCAPE_PLUGIN_API)
+#if defined(XP_X11) && ENABLE(NETSCAPE_PLUGIN_API)
     , m_hasPendingGeometryChange(true)
     , m_drawable(0)
     , m_visual(0)
--- ./Source/WebCore/plugins/PluginView.h~	2011-09-26 13:55:00.000000000 -0700
+++ ./Source/WebCore/plugins/PluginView.h	2012-11-01 15:15:34.000000000 -0700
@@ -408,7 +408,7 @@
         Point mousePosForPlugin(MouseEvent* event = 0) const;
 #endif
 
-#if defined(XP_UNIX) && ENABLE(NETSCAPE_PLUGIN_API)
+#if defined(XP_X11) && ENABLE(NETSCAPE_PLUGIN_API)
         bool m_hasPendingGeometryChange;
         Pixmap m_drawable;
         Visual* m_visual;
--- ./Source/WebKit/gtk/tests/testapplicationcache.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testapplicationcache.c	2012-11-02 13:31:27.000000000 -0700
@@ -39,7 +39,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testatk.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testatk.c	2012-11-02 13:40:25.000000000 -0700
@@ -1660,7 +1660,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(0);
+/*    g_thread_init(0); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, 0);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testatkroles.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testatkroles.c	2012-11-02 13:40:40.000000000 -0700
@@ -302,7 +302,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testcopyandpaste.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testcopyandpaste.c	2012-11-02 13:41:07.000000000 -0700
@@ -228,7 +228,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testdomdocument.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testdomdocument.c	2012-11-02 13:41:37.000000000 -0700
@@ -319,8 +319,8 @@
 
 int main(int argc, char** argv)
 {
-    if (!g_thread_supported())
-        g_thread_init(NULL);
+/*    if (!g_thread_supported())
+        g_thread_init(NULL); Not required after glib-2.32 */
 
     gtk_test_init(&argc, &argv, NULL);
 
--- ./Source/WebKit/gtk/tests/testdomdomwindow.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testdomdomwindow.c	2012-11-02 13:42:00.000000000 -0700
@@ -197,8 +197,8 @@
 
 int main(int argc, char** argv)
 {
-    if (!g_thread_supported())
-        g_thread_init(NULL);
+/*    if (!g_thread_supported())
+        g_thread_init(NULL); Not required after glib-2.32 */
 
     gtk_test_init(&argc, &argv, NULL);
 
--- ./Source/WebKit/gtk/tests/testdomnode.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testdomnode.c	2012-11-02 13:42:23.000000000 -0700
@@ -174,8 +174,8 @@
 
 int main(int argc, char** argv)
 {
-    if (!g_thread_supported())
-        g_thread_init(NULL);
+/*    if (!g_thread_supported())
+        g_thread_init(NULL); Not required after glib-2.32 */
 
     gtk_test_init(&argc, &argv, NULL);
 
--- ./Source/WebKit/gtk/tests/testdownload.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testdownload.c	2012-11-02 13:42:42.000000000 -0700
@@ -250,7 +250,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testglobals.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testglobals.c	2012-11-02 13:43:01.000000000 -0700
@@ -47,7 +47,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testhittestresult.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testhittestresult.c	2012-11-02 13:43:20.000000000 -0700
@@ -136,7 +136,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testhttpbackend.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testhttpbackend.c	2012-11-02 13:43:47.000000000 -0700
@@ -68,7 +68,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testkeyevents.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testkeyevents.c	2012-11-02 13:44:45.000000000 -0700
@@ -303,7 +303,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testloading.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testloading.c	2012-11-02 13:45:04.000000000 -0700
@@ -400,7 +400,7 @@
 {
     SoupServer* server;
 
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     server = soup_server_new(SOUP_SERVER_PORT, 0, NULL);
--- ./Source/WebKit/gtk/tests/testmimehandling.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testmimehandling.c	2012-11-02 13:45:22.000000000 -0700
@@ -176,7 +176,7 @@
     SoupServer* server;
     SoupURI* soup_uri;
 
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     /* Hopefully make test independent of the path it's called from. */
--- ./Source/WebKit/gtk/tests/testnetworkrequest.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testnetworkrequest.c	2012-11-02 13:45:41.000000000 -0700
@@ -88,7 +88,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testnetworkresponse.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testnetworkresponse.c	2012-11-02 13:45:53.000000000 -0700
@@ -89,7 +89,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testwebbackforwardlist.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testwebbackforwardlist.c	2012-11-02 13:47:05.000000000 -0700
@@ -316,7 +316,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testwebdatasource.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testwebdatasource.c	2012-11-02 13:47:21.000000000 -0700
@@ -226,7 +226,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testwebframe.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testwebframe.c	2012-11-02 13:47:37.000000000 -0700
@@ -209,7 +209,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testwebhistoryitem.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testwebhistoryitem.c	2012-11-02 13:47:55.000000000 -0700
@@ -58,7 +58,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testwebplugindatabase.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testwebplugindatabase.c	2012-11-02 13:48:20.000000000 -0700
@@ -71,7 +71,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testwebresource.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testwebresource.c	2012-11-02 13:48:37.000000000 -0700
@@ -295,7 +295,7 @@
     SoupServer* server;
     SoupURI* soup_uri;
 
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     server = soup_server_new(SOUP_SERVER_PORT, 0, NULL);
--- ./Source/WebKit/gtk/tests/testwebsettings.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testwebsettings.c	2012-11-02 13:48:52.000000000 -0700
@@ -149,7 +149,7 @@
 
 int main(int argc, char **argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Source/WebKit/gtk/tests/testwebview.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testwebview.c	2012-11-02 13:49:09.000000000 -0700
@@ -368,7 +368,7 @@
     SoupServer* server;
     SoupURI* soup_uri;
 
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     /* Hopefully make test independent of the path it's called from. */
--- ./Source/WebKit/gtk/tests/testwindow.c~	2011-09-26 13:55:01.000000000 -0700
+++ ./Source/WebKit/gtk/tests/testwindow.c	2012-11-02 13:49:24.000000000 -0700
@@ -112,7 +112,7 @@
 
 int main(int argc, char** argv)
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_test_init(&argc, &argv, NULL);
 
     g_test_bug_base("https://bugs.webkit.org/");
--- ./Tools/DumpRenderTree/gtk/DumpRenderTree.cpp~	2011-09-26 13:55:03.000000000 -0700
+++ ./Tools/DumpRenderTree/gtk/DumpRenderTree.cpp	2012-11-02 14:00:53.000000000 -0700
@@ -1168,7 +1168,7 @@
 
 int main(int argc, char* argv[])
 {
-    g_thread_init(NULL);
+/*    g_thread_init(NULL); Not required after glib-2.32 */
     gtk_init(&argc, &argv);
 
     // Some plugins might try to use the GLib logger for printing debug
--- ./Tools/GtkLauncher/main.c~	2011-09-26 13:55:03.000000000 -0700
+++ ./Tools/GtkLauncher/main.c	2012-11-02 14:01:23.000000000 -0700
@@ -388,8 +388,8 @@
     };
 
     gtk_init(&argc, &argv);
-    if (!g_thread_supported())
-        g_thread_init(NULL);
+/*    if (!g_thread_supported())
+        g_thread_init(NULL);  Not required after glib-2.32 */
 
     GOptionContext *context = g_option_context_new(0);
     g_option_context_add_main_entries(context, commandLineOptions, 0);
--- ./Tools/MiniBrowser/gtk/main.c~	2011-09-26 13:55:03.000000000 -0700
+++ ./Tools/MiniBrowser/gtk/main.c	2012-11-02 14:01:52.000000000 -0700
@@ -76,8 +76,8 @@
 
 int main(int argc, char *argv[])
 {
-    if (!g_thread_supported())
-        g_thread_init(NULL);
+/*    if (!g_thread_supported())
+        g_thread_init(NULL); Not required after glib-2.32 */
 
     GOptionContext *context = g_option_context_new(NULL);
     g_option_context_add_main_entries(context, commandLineOptions, 0);
