--- a/Source/WebKit/gtk/po/GNUmakefile.am	2014-02-15 16:45:30.000000000 -0800
+++ b/Source/WebKit/gtk/po/GNUmakefile.am	2014-02-24 12:21:01.000000000 -0800
@@ -30,7 +30,6 @@
 MOFILES += $(shell echo $(POFILES) | tr ' ' '\n' | sed "s,^$(srcdir)/,,g" | sed 's/\.po/.mo/g')

 .po.mo:
-	test -d Source/WebKit/gtk/po/ || mkdir -p Source/WebKit/gtk/po/
 	@echo "$(MSGFMT) -c -o $@ $<"; \
 	$(MSGFMT) -c -o $@ $<

@@ -60,6 +59,7 @@
 # This target rebuilds $(DOMAIN).pot; it is an expensive operation.
 # Note that $(DOMAIN).pot is not touched if it doesn't need to be changed.
 $(DOMAIN).pot-update: $(POTFILES) $(srcdir)/Source/WebKit/gtk/po/POTFILES
+	test -d $(top_builddir)/Source/WebKit/gtk/po/ || mkdir -p $(top_builddir)/Source/WebKit/gtk/po/
 	package_gnu=''; \
 	if test -n '$(MSGID_BUGS_ADDRESS)' || test '$(PACKAGE_BUGREPORT)' = '@'PACKAGE_BUGREPORT'@'; then \
 	  msgid_bugs_address='$(MSGID_BUGS_ADDRESS)'; \
