diff --git a/GNUmakefile.am b/GNUmakefile.am
index 23cd7ea..d4b19f4 100644
--- a/GNUmakefile.am
+++ b/GNUmakefile.am
@@ -107,9 +107,14 @@ corekit_cppflags += \
 	$(javascriptcore_cppflags)
 
 # For the Gtk port we want to use XP_UNIX both in X11 and Mac
-if !TARGET_WIN32
+if !TARGET_WIN32 
+if TARGET_QUARTZ
+corekit_cppflags += -DXP_MACOSX
+else
 corekit_cppflags += -DXP_UNIX
 endif
+endif
+
 
 # Default compiler flags
 global_cflags += \
diff --git a/GNUmakefile.in b/GNUmakefile.in
index 51bf920..e5d2700 100644
--- a/GNUmakefile.in
+++ b/GNUmakefile.in
@@ -88,8 +88,9 @@ noinst_PROGRAMS = Programs/minidom$(EXEEXT) \
 	$(am__EXEEXT_1)
 
 # For the Gtk port we want to use XP_UNIX both in X11 and Mac
-@TARGET_WIN32_FALSE@am__append_1 = -DXP_UNIX
-@USE_ICU_UNICODE_TRUE@am__append_2 = \
+@TARGET_QUARTZ_TRUE@@TARGET_WIN32_FALSE@am__append_1 = -DXP_MACOSX
+@TARGET_QUARTZ_FALSE@@TARGET_WIN32_FALSE@am__append_2 = -DXP_UNIX
+@USE_ICU_UNICODE_TRUE@am__append_3 = \
 @USE_ICU_UNICODE_TRUE@	-DWTF_USE_ICU_UNICODE=1
 
 
@@ -100,16 +101,16 @@ noinst_PROGRAMS = Programs/minidom$(EXEEXT) \
 # This macro is required for compiling WTF with GLib Unicode backend,
 # but at the same time keeping ICU dependency for codecs and break iterator.
 # It will be removed with patch 3/4 of the above bug.
-@USE_GLIB_UNICODE_TRUE@am__append_3 = \
+@USE_GLIB_UNICODE_TRUE@am__append_4 = \
 @USE_GLIB_UNICODE_TRUE@	-DWTF_USE_GLIB_UNICODE=1 \
 @USE_GLIB_UNICODE_TRUE@	-DWTF_USE_GLIB_ICU_UNICODE_HYBRID=1
 
-@ENABLE_FAST_MALLOC_FALSE@am__append_4 = \
+@ENABLE_FAST_MALLOC_FALSE@am__append_5 = \
 @ENABLE_FAST_MALLOC_FALSE@	-DUSE_SYSTEM_MALLOC
 
-@ENABLE_DEBUG_FALSE@am__append_5 = -DNDEBUG
-@ENABLE_DEBUG_FALSE@am__append_6 = $(SYMBOL_VISIBILITY)
-@ENABLE_DEBUG_TRUE@am__append_7 = \
+@ENABLE_DEBUG_FALSE@am__append_6 = -DNDEBUG
+@ENABLE_DEBUG_FALSE@am__append_7 = $(SYMBOL_VISIBILITY)
+@ENABLE_DEBUG_TRUE@am__append_8 = \
 @ENABLE_DEBUG_TRUE@	-DG_DISABLE_DEPRECATED \
 @ENABLE_DEBUG_TRUE@	-DGDK_PIXBUF_DISABLE_DEPRECATED \
 @ENABLE_DEBUG_TRUE@	-DGDK_DISABLE_DEPRECATED \
@@ -120,7 +121,7 @@ noinst_PROGRAMS = Programs/minidom$(EXEEXT) \
 # Might be useful in the future
 #	-DGDK_MULTIHEAD_SAFE \
 #	-DGTK_MULTIHEAD_SAFE
-@ENABLE_COVERAGE_TRUE@am__append_8 = \
+@ENABLE_COVERAGE_TRUE@am__append_9 = \
 @ENABLE_COVERAGE_TRUE@	-DGCC_GENERATE_TEST_COVERAGE_FILES \
 @ENABLE_COVERAGE_TRUE@	-DGCC_INSTRUMENT_PROGRAM_FLOW_ARCS
 
@@ -139,12 +140,12 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 	$(top_srcdir)/configure ChangeLog INSTALL autotools/compile \
 	autotools/config.guess autotools/config.sub autotools/depcomp \
 	autotools/install-sh autotools/ltmain.sh autotools/missing
-@TARGET_WIN32_TRUE@am__append_9 = \
+@TARGET_WIN32_TRUE@am__append_10 = \
 @TARGET_WIN32_TRUE@	JavaScriptCore/wtf/ThreadSpecificWin.cpp \
 @TARGET_WIN32_TRUE@	JavaScriptCore/jit/ExecutableAllocatorWin.cpp \
 @TARGET_WIN32_TRUE@	JavaScriptCore/runtime/MarkStackWin.cpp
 
-@TARGET_WIN32_FALSE@am__append_10 = \
+@TARGET_WIN32_FALSE@am__append_11 = \
 @TARGET_WIN32_FALSE@	JavaScriptCore/jit/ExecutableAllocatorPosix.cpp \
 @TARGET_WIN32_FALSE@	JavaScriptCore/runtime/MarkStackPosix.cpp
 
@@ -152,7 +153,7 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 # ----
 # icu unicode backend
 # ----
-@USE_ICU_UNICODE_TRUE@am__append_11 = \
+@USE_ICU_UNICODE_TRUE@am__append_12 = \
 @USE_ICU_UNICODE_TRUE@	JavaScriptCore/wtf/unicode/icu/CollatorICU.cpp \
 @USE_ICU_UNICODE_TRUE@	JavaScriptCore/wtf/unicode/icu/UnicodeIcu.h
 
@@ -160,18 +161,18 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 # ----
 # glib unicode backend
 # ----
-@USE_GLIB_UNICODE_TRUE@am__append_12 = \
+@USE_GLIB_UNICODE_TRUE@am__append_13 = \
 @USE_GLIB_UNICODE_TRUE@	JavaScriptCore/wtf/unicode/glib/UnicodeGLib.h \
 @USE_GLIB_UNICODE_TRUE@	JavaScriptCore/wtf/unicode/glib/UnicodeGLib.cpp \
 @USE_GLIB_UNICODE_TRUE@	JavaScriptCore/wtf/unicode/glib/UnicodeMacrosFromICU.h
 
 
 # Debug build
-@ENABLE_DEBUG_TRUE@am__append_13 = \
+@ENABLE_DEBUG_TRUE@am__append_14 = \
 @ENABLE_DEBUG_TRUE@	DerivedSources/Grammar.cpp \
 @ENABLE_DEBUG_TRUE@	DerivedSources/Grammar.h
 
-@ENABLE_DEBUG_TRUE@am__append_14 = \
+@ENABLE_DEBUG_TRUE@am__append_15 = \
 @ENABLE_DEBUG_TRUE@	JavaScriptCore/interpreter/RegisterFile.cpp \
 @ENABLE_DEBUG_TRUE@	JavaScriptCore/interpreter/RegisterFile.h \
 @ENABLE_DEBUG_TRUE@	JavaScriptCore/bytecompiler/BytecodeGenerator.cpp \
@@ -340,21 +341,21 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 @ENABLE_DEBUG_TRUE@	JavaScriptCore/wtf/dtoa.cpp \
 @ENABLE_DEBUG_TRUE@	JavaScriptCore/wtf/dtoa.h
 
-@ENABLE_DEBUG_FALSE@am__append_15 = \
+@ENABLE_DEBUG_FALSE@am__append_16 = \
 @ENABLE_DEBUG_FALSE@	DerivedSources/Grammar.cpp \
 @ENABLE_DEBUG_FALSE@	DerivedSources/Grammar.h
 
-@ENABLE_DEBUG_FALSE@am__append_16 = \
+@ENABLE_DEBUG_FALSE@am__append_17 = \
 @ENABLE_DEBUG_FALSE@	JavaScriptCore/AllInOneFile.cpp \
 @ENABLE_DEBUG_FALSE@	JavaScriptCore/parser/ParserArena.cpp \
 @ENABLE_DEBUG_FALSE@	JavaScriptCore/parser/ParserArena.h
 
-@TARGET_X11_TRUE@am__append_17 = \
+@TARGET_X11_TRUE@am__append_18 = \
 @TARGET_X11_TRUE@	WebCore/plugins/gtk/gtk2xtbin.c \
 @TARGET_X11_TRUE@	WebCore/plugins/gtk/gtk2xtbin.h \
 @TARGET_X11_TRUE@	WebCore/plugins/gtk/xembed.h
 
-@TARGET_WIN32_TRUE@am__append_18 = \
+@TARGET_WIN32_TRUE@am__append_19 = \
 @TARGET_WIN32_TRUE@	WebCore/plugins/win/PluginDatabaseWin.cpp \
 @TARGET_WIN32_TRUE@	WebCore/plugins/win/PluginMessageThrottlerWin.cpp \
 @TARGET_WIN32_TRUE@	WebCore/plugins/win/PluginMessageThrottlerWin.h
@@ -363,18 +364,18 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 # ---
 # Channel mesaging support
 # ---
-@ENABLE_CHANNEL_MESSAGING_TRUE@am__append_19 = ENABLE_CHANNEL_MESSAGING=1
-@ENABLE_CHANNEL_MESSAGING_TRUE@am__append_20 = \
+@ENABLE_CHANNEL_MESSAGING_TRUE@am__append_20 = ENABLE_CHANNEL_MESSAGING=1
+@ENABLE_CHANNEL_MESSAGING_TRUE@am__append_21 = \
 @ENABLE_CHANNEL_MESSAGING_TRUE@	-DENABLE_CHANNEL_MESSAGING=1
 
 
 # ---
 # Freetype font backend
 # ---
-@USE_FREETYPE_TRUE@am__append_21 = \
+@USE_FREETYPE_TRUE@am__append_22 = \
 @USE_FREETYPE_TRUE@	-DUSE_FREETYPE=1
 
-@USE_FREETYPE_TRUE@am__append_22 = \
+@USE_FREETYPE_TRUE@am__append_23 = \
 @USE_FREETYPE_TRUE@	WebCore/platform/graphics/gtk/FontCustomPlatformData.cpp \
 @USE_FREETYPE_TRUE@	WebCore/platform/graphics/gtk/FontPlatformDataGtk.cpp \
 @USE_FREETYPE_TRUE@	WebCore/platform/graphics/gtk/GlyphPageTreeNodeGtk.cpp \
@@ -384,10 +385,10 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 # ---
 # Pango font backend
 # ---
-@USE_PANGO_TRUE@am__append_23 = \
+@USE_PANGO_TRUE@am__append_24 = \
 @USE_PANGO_TRUE@	-DUSE_PANGO=1
 
-@USE_PANGO_TRUE@am__append_24 = \
+@USE_PANGO_TRUE@am__append_25 = \
 @USE_PANGO_TRUE@	WebCore/platform/graphics/gtk/FontCustomPlatformDataPango.cpp \
 @USE_PANGO_TRUE@	WebCore/platform/graphics/gtk/FontPlatformDataPango.cpp \
 @USE_PANGO_TRUE@	WebCore/platform/graphics/gtk/GlyphPageTreeNodePango.cpp \
@@ -397,22 +398,22 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 # ----
 # JavaScript Debugger/Profiler
 # ----
-@ENABLE_JAVASCRIPT_DEBUGGER_TRUE@am__append_25 = ENABLE_JAVASCRIPT_DEBUGGER=1
-@ENABLE_JAVASCRIPT_DEBUGGER_TRUE@am__append_26 = \
+@ENABLE_JAVASCRIPT_DEBUGGER_TRUE@am__append_26 = ENABLE_JAVASCRIPT_DEBUGGER=1
+@ENABLE_JAVASCRIPT_DEBUGGER_TRUE@am__append_27 = \
 @ENABLE_JAVASCRIPT_DEBUGGER_TRUE@	-DENABLE_JAVASCRIPT_DEBUGGER=1
 
 
 # ----
 # Offline Web Applications
 # ----
-@ENABLE_OFFLINE_WEB_APPLICATIONS_TRUE@am__append_27 = ENABLE_OFFLINE_WEB_APPLICATIONS=1
-@ENABLE_OFFLINE_WEB_APPLICATIONS_TRUE@am__append_28 = \
+@ENABLE_OFFLINE_WEB_APPLICATIONS_TRUE@am__append_28 = ENABLE_OFFLINE_WEB_APPLICATIONS=1
+@ENABLE_OFFLINE_WEB_APPLICATIONS_TRUE@am__append_29 = \
 @ENABLE_OFFLINE_WEB_APPLICATIONS_TRUE@	WebCore/loader/appcache/DOMApplicationCache.idl
 
-@ENABLE_OFFLINE_WEB_APPLICATIONS_TRUE@am__append_29 = \
+@ENABLE_OFFLINE_WEB_APPLICATIONS_TRUE@am__append_30 = \
 @ENABLE_OFFLINE_WEB_APPLICATIONS_TRUE@	-DENABLE_OFFLINE_WEB_APPLICATIONS=1
 
-@ENABLE_OFFLINE_WEB_APPLICATIONS_TRUE@am__append_30 = \
+@ENABLE_OFFLINE_WEB_APPLICATIONS_TRUE@am__append_31 = \
 @ENABLE_OFFLINE_WEB_APPLICATIONS_TRUE@	WebCore/bindings/js/JSDOMApplicationCacheCustom.cpp \
 @ENABLE_OFFLINE_WEB_APPLICATIONS_TRUE@	WebCore/loader/appcache/ApplicationCache.cpp \
 @ENABLE_OFFLINE_WEB_APPLICATIONS_TRUE@	WebCore/loader/appcache/ApplicationCache.h \
@@ -433,28 +434,28 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 # ----
 # Dashboard Support
 # ----
-@ENABLE_DASHBOARD_SUPPORT_TRUE@am__append_31 = \
+@ENABLE_DASHBOARD_SUPPORT_TRUE@am__append_32 = \
 @ENABLE_DASHBOARD_SUPPORT_TRUE@	-DENABLE_DASHBOARD_SUPPORT=1
 
-@ENABLE_DASHBOARD_SUPPORT_TRUE@am__append_32 = $(WebCore)/css/DashboardSupportCSSPropertyNames.in
+@ENABLE_DASHBOARD_SUPPORT_TRUE@am__append_33 = $(WebCore)/css/DashboardSupportCSSPropertyNames.in
 
 # ----
 # Database Support
 # ----
-@ENABLE_DATABASE_TRUE@am__append_33 = ENABLE_DATABASE=1
-@ENABLE_DATABASE_TRUE@am__append_34 = \
+@ENABLE_DATABASE_TRUE@am__append_34 = ENABLE_DATABASE=1
+@ENABLE_DATABASE_TRUE@am__append_35 = \
 @ENABLE_DATABASE_TRUE@	WebCore/storage/Database.idl \
 @ENABLE_DATABASE_TRUE@	WebCore/storage/SQLError.idl \
 @ENABLE_DATABASE_TRUE@	WebCore/storage/SQLResultSet.idl \
 @ENABLE_DATABASE_TRUE@	WebCore/storage/SQLResultSetRowList.idl \
 @ENABLE_DATABASE_TRUE@	WebCore/storage/SQLTransaction.idl
 
-@ENABLE_DATABASE_TRUE@am__append_35 = \
+@ENABLE_DATABASE_TRUE@am__append_36 = \
 @ENABLE_DATABASE_TRUE@	-DENABLE_DATABASE=1 \
 @ENABLE_DATABASE_TRUE@	-I$(top_srcdir)/WebCore/platform/sql \
 @ENABLE_DATABASE_TRUE@	-I$(top_srcdir)/WebCore/storage
 
-@ENABLE_DATABASE_TRUE@am__append_36 = \
+@ENABLE_DATABASE_TRUE@am__append_37 = \
 @ENABLE_DATABASE_TRUE@	WebCore/bindings/js/JSCustomSQLStatementCallback.cpp \
 @ENABLE_DATABASE_TRUE@	WebCore/bindings/js/JSCustomSQLStatementCallback.h \
 @ENABLE_DATABASE_TRUE@	WebCore/bindings/js/JSCustomSQLStatementErrorCallback.cpp \
@@ -513,37 +514,37 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 @ENABLE_DATABASE_TRUE@	WebCore/storage/SQLTransactionCoordinator.cpp \
 @ENABLE_DATABASE_TRUE@	WebCore/storage/SQLTransactionErrorCallback.h
 
-@ENABLE_DATABASE_FALSE@am__append_37 = -DENABLE_DATABASE=0
+@ENABLE_DATABASE_FALSE@am__append_38 = -DENABLE_DATABASE=0
 
 # ----
 # HTML5 datalist Support
 # ----
-@ENABLE_DATALIST_TRUE@am__append_38 = ENABLE_DATALIST=1
 @ENABLE_DATALIST_TRUE@am__append_39 = ENABLE_DATALIST=1
-@ENABLE_DATALIST_TRUE@am__append_40 = -DENABLE_DATALIST=1
+@ENABLE_DATALIST_TRUE@am__append_40 = ENABLE_DATALIST=1
+@ENABLE_DATALIST_TRUE@am__append_41 = -DENABLE_DATALIST=1
 
 # ----
 # HTML5 server-sent events
 # ----
-@ENABLE_EVENTSOURCE_FALSE@am__append_41 = -DENABLE_EVENTSOURCE=0
-@ENABLE_EVENTSOURCE_TRUE@am__append_42 = ENABLE_EVENTSOURCE=1
-@ENABLE_EVENTSOURCE_TRUE@am__append_43 = \
+@ENABLE_EVENTSOURCE_FALSE@am__append_42 = -DENABLE_EVENTSOURCE=0
+@ENABLE_EVENTSOURCE_TRUE@am__append_43 = ENABLE_EVENTSOURCE=1
+@ENABLE_EVENTSOURCE_TRUE@am__append_44 = \
 @ENABLE_EVENTSOURCE_TRUE@	-DENABLE_EVENTSOURCE=1
 
 
 # ----
 # HTML5 client-side session and persistent storage
 # ----
-@ENABLE_DOM_STORAGE_TRUE@am__append_44 = ENABLE_DOM_STORAGE=1
-@ENABLE_DOM_STORAGE_TRUE@am__append_45 = \
+@ENABLE_DOM_STORAGE_TRUE@am__append_45 = ENABLE_DOM_STORAGE=1
+@ENABLE_DOM_STORAGE_TRUE@am__append_46 = \
 @ENABLE_DOM_STORAGE_TRUE@	WebCore/storage/Storage.idl \
 @ENABLE_DOM_STORAGE_TRUE@	WebCore/storage/StorageEvent.idl
 
-@ENABLE_DOM_STORAGE_TRUE@am__append_46 = \
+@ENABLE_DOM_STORAGE_TRUE@am__append_47 = \
 @ENABLE_DOM_STORAGE_TRUE@	-DENABLE_DOM_STORAGE=1 \
 @ENABLE_DOM_STORAGE_TRUE@	-I$(top_srcdir)/WebCore/storage
 
-@ENABLE_DOM_STORAGE_TRUE@am__append_47 = \
+@ENABLE_DOM_STORAGE_TRUE@am__append_48 = \
 @ENABLE_DOM_STORAGE_TRUE@	WebCore/bindings/js/JSStorageCustom.cpp \
 @ENABLE_DOM_STORAGE_TRUE@	WebCore/storage/LocalStorageTask.cpp \
 @ENABLE_DOM_STORAGE_TRUE@	WebCore/storage/LocalStorageTask.h \
@@ -571,8 +572,8 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 # ----
 # Icon Database Support
 # ----
-@ENABLE_ICONDATABASE_TRUE@am__append_48 = -DENABLE_ICONDATABASE=1
-@ENABLE_ICONDATABASE_TRUE@am__append_49 = \
+@ENABLE_ICONDATABASE_TRUE@am__append_49 = -DENABLE_ICONDATABASE=1
+@ENABLE_ICONDATABASE_TRUE@am__append_50 = \
 @ENABLE_ICONDATABASE_TRUE@	WebCore/loader/icon/IconDatabase.cpp \
 @ENABLE_ICONDATABASE_TRUE@	WebCore/loader/icon/IconDatabase.h \
 @ENABLE_ICONDATABASE_TRUE@	WebCore/loader/icon/IconDatabaseClient.h \
@@ -581,22 +582,22 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 @ENABLE_ICONDATABASE_TRUE@	WebCore/loader/icon/PageURLRecord.cpp \
 @ENABLE_ICONDATABASE_TRUE@	WebCore/loader/icon/PageURLRecord.h
 
-@ENABLE_ICONDATABASE_FALSE@am__append_50 = \
+@ENABLE_ICONDATABASE_FALSE@am__append_51 = \
 @ENABLE_ICONDATABASE_FALSE@	WebCore/loader/icon/IconDatabaseNone.cpp
 
-@ENABLE_ICONDATABASE_FALSE@am__append_51 = -DENABLE_ICONDATABASE=0
+@ENABLE_ICONDATABASE_FALSE@am__append_52 = -DENABLE_ICONDATABASE=0
 
 # ----
 # Video Support
 # ----
-@ENABLE_VIDEO_TRUE@am__append_52 = ENABLE_VIDEO=1
 @ENABLE_VIDEO_TRUE@am__append_53 = ENABLE_VIDEO=1
-@ENABLE_VIDEO_TRUE@am__append_54 = -DENABLE_VIDEO=1
-@ENABLE_DEBUG_TRUE@@ENABLE_VIDEO_TRUE@am__append_55 = -DGST_DISABLE_DEPRECATED
-@ENABLE_VIDEO_TRUE@am__append_56 = \
+@ENABLE_VIDEO_TRUE@am__append_54 = ENABLE_VIDEO=1
+@ENABLE_VIDEO_TRUE@am__append_55 = -DENABLE_VIDEO=1
+@ENABLE_DEBUG_TRUE@@ENABLE_VIDEO_TRUE@am__append_56 = -DGST_DISABLE_DEPRECATED
+@ENABLE_VIDEO_TRUE@am__append_57 = \
 @ENABLE_VIDEO_TRUE@	WebCore/html/TimeRanges.idl
 
-@ENABLE_VIDEO_TRUE@am__append_57 = \
+@ENABLE_VIDEO_TRUE@am__append_58 = \
 @ENABLE_VIDEO_TRUE@	WebCore/bindings/js/JSAudioConstructor.cpp \
 @ENABLE_VIDEO_TRUE@	WebCore/bindings/js/JSAudioConstructor.h \
 @ENABLE_VIDEO_TRUE@	WebCore/html/HTMLAudioElement.cpp \
@@ -617,7 +618,7 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 @ENABLE_VIDEO_TRUE@	WebCore/rendering/RenderVideo.cpp \
 @ENABLE_VIDEO_TRUE@	WebCore/rendering/RenderVideo.h
 
-@ENABLE_VIDEO_TRUE@am__append_58 = \
+@ENABLE_VIDEO_TRUE@am__append_59 = \
 @ENABLE_VIDEO_TRUE@	WebCore/platform/graphics/gtk/MediaPlayerPrivateGStreamer.cpp \
 @ENABLE_VIDEO_TRUE@	WebCore/platform/graphics/gtk/MediaPlayerPrivateGStreamer.h \
 @ENABLE_VIDEO_TRUE@	WebCore/platform/graphics/gtk/VideoSinkGStreamer.cpp \
@@ -625,7 +626,7 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 @ENABLE_VIDEO_TRUE@	WebCore/platform/graphics/gtk/DataSourceGStreamer.cpp \
 @ENABLE_VIDEO_TRUE@	WebCore/platform/graphics/gtk/DataSourceGStreamer.h
 
-@ENABLE_VIDEO_TRUE@am__append_59 = \
+@ENABLE_VIDEO_TRUE@am__append_60 = \
 @ENABLE_VIDEO_TRUE@	-lgstinterfaces-0.10 \
 @ENABLE_VIDEO_TRUE@	-lgstvideo-0.10 \
 @ENABLE_VIDEO_TRUE@	-lgstpbutils-0.10
@@ -634,26 +635,26 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 # ----
 # Ruby Support
 # ----
-@ENABLE_RUBY_TRUE@am__append_60 = ENABLE_RUBY=1
-@ENABLE_RUBY_TRUE@am__append_61 = -DENABLE_RUBY=1
+@ENABLE_RUBY_TRUE@am__append_61 = ENABLE_RUBY=1
+@ENABLE_RUBY_TRUE@am__append_62 = -DENABLE_RUBY=1
 
 # ----
 # XPath Support
 # ----
-@ENABLE_XPATH_TRUE@am__append_62 = ENABLE_XPATH=1
-@ENABLE_XPATH_TRUE@am__append_63 = -DENABLE_XPATH=1
-@ENABLE_XPATH_TRUE@am__append_64 = \
+@ENABLE_XPATH_TRUE@am__append_63 = ENABLE_XPATH=1
+@ENABLE_XPATH_TRUE@am__append_64 = -DENABLE_XPATH=1
+@ENABLE_XPATH_TRUE@am__append_65 = \
 @ENABLE_XPATH_TRUE@	DerivedSources/XPathGrammar.cpp \
 @ENABLE_XPATH_TRUE@	DerivedSources/XPathGrammar.h
 
-@ENABLE_XPATH_TRUE@am__append_65 = \
+@ENABLE_XPATH_TRUE@am__append_66 = \
 @ENABLE_XPATH_TRUE@	WebCore/xml/XPathEvaluator.idl \
 @ENABLE_XPATH_TRUE@	WebCore/xml/XPathException.idl \
 @ENABLE_XPATH_TRUE@	WebCore/xml/XPathExpression.idl \
 @ENABLE_XPATH_TRUE@	WebCore/xml/XPathNSResolver.idl \
 @ENABLE_XPATH_TRUE@	WebCore/xml/XPathResult.idl
 
-@ENABLE_XPATH_TRUE@am__append_66 = \
+@ENABLE_XPATH_TRUE@am__append_67 = \
 @ENABLE_XPATH_TRUE@	WebCore/xml/NativeXPathNSResolver.cpp \
 @ENABLE_XPATH_TRUE@	WebCore/xml/NativeXPathNSResolver.h \
 @ENABLE_XPATH_TRUE@	WebCore/xml/XPathEvaluator.cpp \
@@ -692,9 +693,9 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 # ----
 # XSLT Support
 # ----
-@ENABLE_XSLT_TRUE@am__append_67 = ENABLE_XSLT=1
-@ENABLE_XSLT_TRUE@am__append_68 = -DENABLE_XSLT=1
-@ENABLE_XSLT_TRUE@am__append_69 = \
+@ENABLE_XSLT_TRUE@am__append_68 = ENABLE_XSLT=1
+@ENABLE_XSLT_TRUE@am__append_69 = -DENABLE_XSLT=1
+@ENABLE_XSLT_TRUE@am__append_70 = \
 @ENABLE_XSLT_TRUE@	WebCore/xml/XSLImportRule.cpp \
 @ENABLE_XSLT_TRUE@	WebCore/xml/XSLImportRule.h \
 @ENABLE_XSLT_TRUE@	WebCore/xml/XSLStyleSheet.cpp \
@@ -710,11 +711,11 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 # ----
 # Web Workers support
 # ----
-@ENABLE_WORKERS_TRUE@am__append_70 = ENABLE_WORKERS=1
-@ENABLE_WORKERS_TRUE@am__append_71 = \
+@ENABLE_WORKERS_TRUE@am__append_71 = ENABLE_WORKERS=1
+@ENABLE_WORKERS_TRUE@am__append_72 = \
 @ENABLE_WORKERS_TRUE@	-DENABLE_WORKERS=1
 
-@ENABLE_WORKERS_TRUE@am__append_72 = \
+@ENABLE_WORKERS_TRUE@am__append_73 = \
 @ENABLE_WORKERS_TRUE@	WebCore/bindings/js/JSAbstractWorkerCustom.cpp \
 @ENABLE_WORKERS_TRUE@	WebCore/bindings/js/JSDedicatedWorkerContextCustom.cpp \
 @ENABLE_WORKERS_TRUE@	WebCore/bindings/js/JSWorkerConstructor.cpp \
@@ -758,11 +759,11 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 # ---
 # SharedWorker support
 # ---
-@ENABLE_SHARED_WORKERS_TRUE@am__append_73 = ENABLE_SHARED_WORKERS=1
-@ENABLE_SHARED_WORKERS_TRUE@am__append_74 = \
+@ENABLE_SHARED_WORKERS_TRUE@am__append_74 = ENABLE_SHARED_WORKERS=1
+@ENABLE_SHARED_WORKERS_TRUE@am__append_75 = \
 @ENABLE_SHARED_WORKERS_TRUE@	-DENABLE_SHARED_WORKERS=1
 
-@ENABLE_SHARED_WORKERS_TRUE@am__append_75 = \
+@ENABLE_SHARED_WORKERS_TRUE@am__append_76 = \
 @ENABLE_SHARED_WORKERS_TRUE@	WebCore/bindings/js/JSSharedWorkerConstructor.cpp \
 @ENABLE_SHARED_WORKERS_TRUE@	WebCore/bindings/js/JSSharedWorkerConstructor.h \
 @ENABLE_SHARED_WORKERS_TRUE@	WebCore/bindings/js/JSSharedWorkerContextCustom.cpp \
@@ -781,16 +782,16 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 # ----
 # WML Support
 # ----
-@ENABLE_WML_TRUE@am__append_76 = ENABLE_WML=1
-@ENABLE_WML_TRUE@am__append_77 = \
+@ENABLE_WML_TRUE@am__append_77 = ENABLE_WML=1
+@ENABLE_WML_TRUE@am__append_78 = \
 @ENABLE_WML_TRUE@	DerivedSources/WMLElementFactory.cpp \
 @ENABLE_WML_TRUE@	DerivedSources/WMLNames.cpp
 
-@ENABLE_WML_TRUE@am__append_78 = \
+@ENABLE_WML_TRUE@am__append_79 = \
 @ENABLE_WML_TRUE@	-DENABLE_WML=1 \
 @ENABLE_WML_TRUE@	-I$(top_srcdir)/WebCore/wml
 
-@ENABLE_WML_TRUE@am__append_79 = \
+@ENABLE_WML_TRUE@am__append_80 = \
 @ENABLE_WML_TRUE@	WebCore/wml/WMLAElement.cpp \
 @ENABLE_WML_TRUE@	WebCore/wml/WMLAccessElement.cpp \
 @ENABLE_WML_TRUE@	WebCore/wml/WMLAnchorElement.cpp \
@@ -832,9 +833,9 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 # ----
 # Filters
 # ----
-@ENABLE_FILTERS_TRUE@am__append_80 = ENABLE_FILTERS=1
-@ENABLE_FILTERS_TRUE@am__append_81 = -DENABLE_FILTERS=1
-@ENABLE_FILTERS_TRUE@am__append_82 = \
+@ENABLE_FILTERS_TRUE@am__append_81 = ENABLE_FILTERS=1
+@ENABLE_FILTERS_TRUE@am__append_82 = -DENABLE_FILTERS=1
+@ENABLE_FILTERS_TRUE@am__append_83 = \
 @ENABLE_FILTERS_TRUE@	WebCore/platform/graphics/filters/FEBlend.cpp \
 @ENABLE_FILTERS_TRUE@	WebCore/platform/graphics/filters/FEBlend.h \
 @ENABLE_FILTERS_TRUE@	WebCore/platform/graphics/filters/FEColorMatrix.cpp \
@@ -851,16 +852,16 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 @ENABLE_FILTERS_TRUE@	WebCore/platform/graphics/filters/SourceGraphic.cpp \
 @ENABLE_FILTERS_TRUE@	WebCore/platform/graphics/filters/SourceGraphic.h
 
-@ENABLE_FILTERS_TRUE@@ENABLE_SVG_TRUE@am__append_83 = ENABLE_FILTERS=1
+@ENABLE_FILTERS_TRUE@@ENABLE_SVG_TRUE@am__append_84 = ENABLE_FILTERS=1
 
 # ----
 # Geolocation
 # ----
-@ENABLE_GEOLOCATION_TRUE@am__append_84 = ENABLE_GEOLOCATION=1
-@ENABLE_GEOLOCATION_TRUE@am__append_85 = \
+@ENABLE_GEOLOCATION_TRUE@am__append_85 = ENABLE_GEOLOCATION=1
+@ENABLE_GEOLOCATION_TRUE@am__append_86 = \
 @ENABLE_GEOLOCATION_TRUE@	-DENABLE_GEOLOCATION=1
 
-@ENABLE_GEOLOCATION_TRUE@am__append_86 = \
+@ENABLE_GEOLOCATION_TRUE@am__append_87 = \
 @ENABLE_GEOLOCATION_TRUE@	WebCore/platform/gtk/GeolocationServiceGtk.cpp \
 @ENABLE_GEOLOCATION_TRUE@	WebCore/platform/gtk/GeolocationServiceGtk.h
 
@@ -868,11 +869,11 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 # ----
 # Notifications support
 # ----
-@ENABLE_NOTIFICATIONS_TRUE@am__append_87 = ENABLE_NOTIFICATIONS=1
-@ENABLE_NOTIFICATIONS_TRUE@am__append_88 = \
+@ENABLE_NOTIFICATIONS_TRUE@am__append_88 = ENABLE_NOTIFICATIONS=1
+@ENABLE_NOTIFICATIONS_TRUE@am__append_89 = \
 @ENABLE_NOTIFICATIONS_TRUE@	-DENABLE_NOTIFICATIONS=1
 
-@ENABLE_NOTIFICATIONS_TRUE@am__append_89 = \
+@ENABLE_NOTIFICATIONS_TRUE@am__append_90 = \
 @ENABLE_NOTIFICATIONS_TRUE@	WebCore/notifications/Notification.cpp \
 @ENABLE_NOTIFICATIONS_TRUE@	WebCore/notifications/Notification.h \
 @ENABLE_NOTIFICATIONS_TRUE@	WebCore/notifications/NotificationCenter.cpp \
@@ -886,16 +887,16 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 #
 # FIXME: allow a more fine-grained inclusion/generation of sources per SVG feature
 # ----
-@ENABLE_SVG_TRUE@am__append_90 = ENABLE_SVG=1
-@ENABLE_SVG_TRUE@am__append_91 = $(WebCore)/css/SVGCSSPropertyNames.in 
-@ENABLE_SVG_TRUE@am__append_92 = $(WebCore)/css/SVGCSSValueKeywords.in
-@ENABLE_SVG_TRUE@am__append_93 = \
+@ENABLE_SVG_TRUE@am__append_91 = ENABLE_SVG=1
+@ENABLE_SVG_TRUE@am__append_92 = $(WebCore)/css/SVGCSSPropertyNames.in 
+@ENABLE_SVG_TRUE@am__append_93 = $(WebCore)/css/SVGCSSValueKeywords.in
+@ENABLE_SVG_TRUE@am__append_94 = \
 @ENABLE_SVG_TRUE@	DerivedSources/JSSVGElementWrapperFactory.cpp \
 @ENABLE_SVG_TRUE@	DerivedSources/SVGElementFactory.cpp \
 @ENABLE_SVG_TRUE@	DerivedSources/SVGNames.cpp \
 @ENABLE_SVG_TRUE@	DerivedSources/XLinkNames.cpp
 
-@ENABLE_SVG_TRUE@am__append_94 = \
+@ENABLE_SVG_TRUE@am__append_95 = \
 @ENABLE_SVG_TRUE@	WebCore/svg/ElementTimeControl.idl \
 @ENABLE_SVG_TRUE@	WebCore/svg/SVGAElement.idl \
 @ENABLE_SVG_TRUE@	WebCore/svg/SVGAltGlyphElement.idl \
@@ -1043,14 +1044,14 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 @ENABLE_SVG_TRUE@	WebCore/svg/SVGZoomAndPan.idl \
 @ENABLE_SVG_TRUE@	WebCore/svg/SVGZoomEvent.idl
 
-@ENABLE_SVG_TRUE@am__append_95 = \
+@ENABLE_SVG_TRUE@am__append_96 = \
 @ENABLE_SVG_TRUE@	-DENABLE_SVG=1 \
 @ENABLE_SVG_TRUE@	-I$(top_srcdir)/WebCore/svg \
 @ENABLE_SVG_TRUE@	-I$(top_srcdir)/WebCore/svg/animation \
 @ENABLE_SVG_TRUE@	-I$(top_srcdir)/WebCore/svg/graphics \
 @ENABLE_SVG_TRUE@	-I$(top_srcdir)/WebCore/svg/graphics/filters
 
-@ENABLE_SVG_TRUE@am__append_96 = \
+@ENABLE_SVG_TRUE@am__append_97 = \
 @ENABLE_SVG_TRUE@	WebCore/bindings/js/JSSVGElementInstanceCustom.cpp \
 @ENABLE_SVG_TRUE@	WebCore/bindings/js/JSSVGLengthCustom.cpp \
 @ENABLE_SVG_TRUE@	WebCore/bindings/js/JSSVGMatrixCustom.cpp \
@@ -1451,34 +1452,34 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 
 
 # SVG Features
-@ENABLE_SVG_TRUE@@ENABLE_SVG_USE_TRUE@am__append_97 = ENABLE_SVG_USE=1
 @ENABLE_SVG_TRUE@@ENABLE_SVG_USE_TRUE@am__append_98 = ENABLE_SVG_USE=1
-@ENABLE_SVG_TRUE@@ENABLE_SVG_USE_TRUE@am__append_99 = -DENABLE_SVG_USE=1
-@ENABLE_SVG_FOREIGN_OBJECT_TRUE@@ENABLE_SVG_TRUE@am__append_100 = ENABLE_SVG_FOREIGN_OBJECT=1
+@ENABLE_SVG_TRUE@@ENABLE_SVG_USE_TRUE@am__append_99 = ENABLE_SVG_USE=1
+@ENABLE_SVG_TRUE@@ENABLE_SVG_USE_TRUE@am__append_100 = -DENABLE_SVG_USE=1
 @ENABLE_SVG_FOREIGN_OBJECT_TRUE@@ENABLE_SVG_TRUE@am__append_101 = ENABLE_SVG_FOREIGN_OBJECT=1
-@ENABLE_SVG_FOREIGN_OBJECT_TRUE@@ENABLE_SVG_TRUE@am__append_102 = -DENABLE_SVG_FOREIGN_OBJECT=1
-@ENABLE_SVG_FONTS_TRUE@@ENABLE_SVG_TRUE@am__append_103 = ENABLE_SVG_FONTS=1
+@ENABLE_SVG_FOREIGN_OBJECT_TRUE@@ENABLE_SVG_TRUE@am__append_102 = ENABLE_SVG_FOREIGN_OBJECT=1
+@ENABLE_SVG_FOREIGN_OBJECT_TRUE@@ENABLE_SVG_TRUE@am__append_103 = -DENABLE_SVG_FOREIGN_OBJECT=1
 @ENABLE_SVG_FONTS_TRUE@@ENABLE_SVG_TRUE@am__append_104 = ENABLE_SVG_FONTS=1
-@ENABLE_SVG_FONTS_TRUE@@ENABLE_SVG_TRUE@am__append_105 = -DENABLE_SVG_FONTS=1
-@ENABLE_SVG_AS_IMAGE_TRUE@@ENABLE_SVG_TRUE@am__append_106 = ENABLE_SVG_AS_IMAGE=1
+@ENABLE_SVG_FONTS_TRUE@@ENABLE_SVG_TRUE@am__append_105 = ENABLE_SVG_FONTS=1
+@ENABLE_SVG_FONTS_TRUE@@ENABLE_SVG_TRUE@am__append_106 = -DENABLE_SVG_FONTS=1
 @ENABLE_SVG_AS_IMAGE_TRUE@@ENABLE_SVG_TRUE@am__append_107 = ENABLE_SVG_AS_IMAGE=1
-@ENABLE_SVG_AS_IMAGE_TRUE@@ENABLE_SVG_TRUE@am__append_108 = -DENABLE_SVG_AS_IMAGE=1
-@ENABLE_SVG_ANIMATION_TRUE@@ENABLE_SVG_TRUE@am__append_109 = ENABLE_SVG_ANIMATION=1
+@ENABLE_SVG_AS_IMAGE_TRUE@@ENABLE_SVG_TRUE@am__append_108 = ENABLE_SVG_AS_IMAGE=1
+@ENABLE_SVG_AS_IMAGE_TRUE@@ENABLE_SVG_TRUE@am__append_109 = -DENABLE_SVG_AS_IMAGE=1
 @ENABLE_SVG_ANIMATION_TRUE@@ENABLE_SVG_TRUE@am__append_110 = ENABLE_SVG_ANIMATION=1
-@ENABLE_SVG_ANIMATION_TRUE@@ENABLE_SVG_TRUE@am__append_111 = -DENABLE_SVG_ANIMATION=1
+@ENABLE_SVG_ANIMATION_TRUE@@ENABLE_SVG_TRUE@am__append_111 = ENABLE_SVG_ANIMATION=1
+@ENABLE_SVG_ANIMATION_TRUE@@ENABLE_SVG_TRUE@am__append_112 = -DENABLE_SVG_ANIMATION=1
 
 # ----
 # Web Sockets Support
 # ----
-@ENABLE_WEB_SOCKETS_TRUE@am__append_112 = ENABLE_WEB_SOCKETS=1
-@ENABLE_WEB_SOCKETS_TRUE@am__append_113 = \
+@ENABLE_WEB_SOCKETS_TRUE@am__append_113 = ENABLE_WEB_SOCKETS=1
+@ENABLE_WEB_SOCKETS_TRUE@am__append_114 = \
 @ENABLE_WEB_SOCKETS_TRUE@	WebCore/websockets/WebSocket.idl
 
-@ENABLE_WEB_SOCKETS_TRUE@am__append_114 = \
+@ENABLE_WEB_SOCKETS_TRUE@am__append_115 = \
 @ENABLE_WEB_SOCKETS_TRUE@	-DENABLE_WEB_SOCKETS=1 \
 @ENABLE_WEB_SOCKETS_TRUE@	-I$(srcdir)/WebCore/websockets
 
-@ENABLE_WEB_SOCKETS_TRUE@am__append_115 = \
+@ENABLE_WEB_SOCKETS_TRUE@am__append_116 = \
 @ENABLE_WEB_SOCKETS_TRUE@	WebCore/bindings/js/JSWebSocketConstructor.cpp \
 @ENABLE_WEB_SOCKETS_TRUE@	WebCore/bindings/js/JSWebSocketConstructor.h \
 @ENABLE_WEB_SOCKETS_TRUE@	WebCore/bindings/js/JSWebSocketCustom.cpp \
@@ -1495,22 +1496,22 @@ DIST_COMMON = README $(am__configure_deps) $(am__noinst_HEADERS_DIST) \
 @ENABLE_WEB_SOCKETS_TRUE@	WebCore/websockets/WebSocketHandshake.cpp \
 @ENABLE_WEB_SOCKETS_TRUE@	WebCore/websockets/WebSocketHandshake.h
 
-@ENABLE_WEB_SOCKETS_TRUE@am__append_116 = \
+@ENABLE_WEB_SOCKETS_TRUE@am__append_117 = \
 @ENABLE_WEB_SOCKETS_TRUE@	WebCore/platform/network/soup/SocketStreamError.h \
 @ENABLE_WEB_SOCKETS_TRUE@	WebCore/platform/network/soup/SocketStreamHandle.h \
 @ENABLE_WEB_SOCKETS_TRUE@	WebCore/platform/network/soup/SocketStreamHandleSoup.cpp
 
-@ENABLE_WEB_SOCKETS_FALSE@am__append_117 = -DENABLE_WEB_SOCKETS=0
+@ENABLE_WEB_SOCKETS_FALSE@am__append_118 = -DENABLE_WEB_SOCKETS=0
 
 # Build TestNetscapePlugin only for X11
 # since we don't support plugins for non-X11 builds at the moment.
-@TARGET_X11_TRUE@am__append_118 = \
+@TARGET_X11_TRUE@am__append_119 = \
 @TARGET_X11_TRUE@	TestNetscapePlugin/libtestnetscapeplugin.la
 
-@TARGET_X11_TRUE@am__append_119 = \
+@TARGET_X11_TRUE@am__append_120 = \
 @TARGET_X11_TRUE@	-DTEST_PLUGIN_DIR=\"${shell pwd}/${top_builddir}/TestNetscapePlugin/.libs\"
 
-@TARGET_X11_TRUE@am__append_120 = TestNetscapePlugin/libtestnetscapeplugin.la
+@TARGET_X11_TRUE@am__append_121 = TestNetscapePlugin/libtestnetscapeplugin.la
 subdir = .
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/autotools/acinclude.m4 \
@@ -6340,9 +6341,9 @@ IDL_BINDINGS := WebCore/css/CSSCharsetRule.idl \
 	WebCore/xml/XMLHttpRequestProgressEvent.idl \
 	WebCore/xml/XMLHttpRequestUpload.idl \
 	WebCore/xml/XMLSerializer.idl WebCore/xml/XSLTProcessor.idl \
-	$(am__append_28) $(am__append_34) $(am__append_45) \
-	$(am__append_56) $(am__append_65) $(am__append_94) \
-	$(am__append_113)
+	$(am__append_29) $(am__append_35) $(am__append_46) \
+	$(am__append_57) $(am__append_66) $(am__append_95) \
+	$(am__append_114)
 TEST_PROGS := Programs/unittests/testhttpbackend \
 	Programs/unittests/testloading \
 	Programs/unittests/testnetworkrequest \
@@ -6614,8 +6615,8 @@ javascriptcore_sources := JavaScriptCore/API/APICast.h \
 	JavaScriptCore/wtf/unicode/CollatorDefault.cpp \
 	JavaScriptCore/wtf/unicode/UTF8.cpp \
 	JavaScriptCore/wtf/unicode/UTF8.h \
-	JavaScriptCore/wtf/unicode/Unicode.h $(am__append_9) \
-	$(am__append_10) $(am__append_11) $(am__append_12) \
+	JavaScriptCore/wtf/unicode/Unicode.h $(am__append_10) \
+	$(am__append_11) $(am__append_12) $(am__append_13) \
 	JavaScriptCore/wtf/VMTags.h \
 	JavaScriptCore/yarr/RegexCompiler.cpp \
 	JavaScriptCore/yarr/RegexCompiler.h \
@@ -6624,9 +6625,9 @@ javascriptcore_sources := JavaScriptCore/API/APICast.h \
 	JavaScriptCore/yarr/RegexJIT.cpp \
 	JavaScriptCore/yarr/RegexJIT.h \
 	JavaScriptCore/yarr/RegexParser.h \
-	JavaScriptCore/yarr/RegexPattern.h $(am__append_14) \
-	$(am__append_16)
-javascriptcore_built_sources := $(am__append_13)
+	JavaScriptCore/yarr/RegexPattern.h $(am__append_15) \
+	$(am__append_17)
+javascriptcore_built_sources := $(am__append_14)
 javascriptcore_built_nosources := DerivedSources/Lexer.lut.h \
 	JavaScriptCore/runtime/ArrayPrototype.lut.h \
 	JavaScriptCore/runtime/DatePrototype.lut.h \
@@ -6636,7 +6637,7 @@ javascriptcore_built_nosources := DerivedSources/Lexer.lut.h \
 	JavaScriptCore/runtime/RegExpConstructor.lut.h \
 	JavaScriptCore/runtime/RegExpObject.lut.h \
 	JavaScriptCore/runtime/StringPrototype.lut.h \
-	JavaScriptCore/pcre/chartables.c $(am__append_15)
+	JavaScriptCore/pcre/chartables.c $(am__append_16)
 javascriptcore_dist := $(CREATE_HASH_TABLE) JavaScriptCore/AUTHORS \
 	JavaScriptCore/COPYING.LIB JavaScriptCore/ChangeLog \
 	JavaScriptCore/THANKS JavaScriptCore/icu/LICENSE \
@@ -6677,17 +6678,17 @@ webcore_cppflags := -I$(srcdir)/WebCore \
 	-I$(srcdir)/WebCore/rendering/style \
 	-I$(srcdir)/WebCore/workers -I$(srcdir)/WebCore/xml \
 	-I$(top_builddir)/WebCore/bindings/js \
-	-DDATA_DIR=\"${datadir}\" $(am__append_20) -DENABLE_DATAGRID=1 \
-	$(am__append_26) $(am__append_29) $(am__append_31) \
-	$(am__append_35) $(am__append_37) $(am__append_40) \
-	$(am__append_43) $(am__append_46) $(am__append_48) \
-	$(am__append_51) $(am__append_54) $(am__append_55) \
-	$(am__append_61) $(am__append_63) $(am__append_68) \
-	$(am__append_71) $(am__append_74) $(am__append_78) \
-	$(am__append_81) $(am__append_85) $(am__append_88) \
-	$(am__append_95) $(am__append_99) $(am__append_102) \
-	$(am__append_105) $(am__append_108) $(am__append_111) \
-	$(am__append_114) $(am__append_117)
+	-DDATA_DIR=\"${datadir}\" $(am__append_21) -DENABLE_DATAGRID=1 \
+	$(am__append_27) $(am__append_30) $(am__append_32) \
+	$(am__append_36) $(am__append_38) $(am__append_41) \
+	$(am__append_44) $(am__append_47) $(am__append_49) \
+	$(am__append_52) $(am__append_55) $(am__append_56) \
+	$(am__append_62) $(am__append_64) $(am__append_69) \
+	$(am__append_72) $(am__append_75) $(am__append_79) \
+	$(am__append_82) $(am__append_86) $(am__append_89) \
+	$(am__append_96) $(am__append_100) $(am__append_103) \
+	$(am__append_106) $(am__append_109) $(am__append_112) \
+	$(am__append_115) $(am__append_118)
 webcore_sources := WebCore/WebCorePrefix.h \
 	WebCore/accessibility/AXObjectCache.cpp \
 	WebCore/accessibility/AXObjectCache.h \
@@ -8015,12 +8016,12 @@ webcore_sources := WebCore/WebCorePrefix.h \
 	WebCore/xml/XMLHttpRequestUpload.cpp \
 	WebCore/xml/XMLHttpRequestUpload.h \
 	WebCore/xml/XMLSerializer.cpp WebCore/xml/XMLSerializer.h \
-	$(am__append_18) $(am__append_30) $(am__append_36) \
-	$(am__append_47) $(am__append_49) $(am__append_50) \
-	$(am__append_57) $(am__append_66) $(am__append_69) \
-	$(am__append_72) $(am__append_79) $(am__append_82) \
-	$(am__append_89) $(am__append_96) $(am__append_115)
-webcore_libadd := $(am__append_59)
+	$(am__append_19) $(am__append_31) $(am__append_37) \
+	$(am__append_48) $(am__append_50) $(am__append_51) \
+	$(am__append_58) $(am__append_67) $(am__append_70) \
+	$(am__append_73) $(am__append_80) $(am__append_83) \
+	$(am__append_90) $(am__append_97) $(am__append_116)
+webcore_libadd := $(am__append_60)
 webcore_built_sources := DerivedSources/CSSGrammar.cpp \
 	DerivedSources/CSSGrammar.h DerivedSources/CSSPropertyNames.h \
 	DerivedSources/CSSValueKeywords.h \
@@ -8029,8 +8030,8 @@ webcore_built_sources := DerivedSources/CSSGrammar.cpp \
 	DerivedSources/JSHTMLElementWrapperFactory.cpp \
 	DerivedSources/UserAgentStyleSheets.h \
 	DerivedSources/UserAgentStyleSheetsData.cpp \
-	DerivedSources/XMLNames.cpp $(am__append_64) $(am__append_75) \
-	$(am__append_77) $(am__append_93)
+	DerivedSources/XMLNames.cpp $(am__append_65) $(am__append_76) \
+	$(am__append_78) $(am__append_94)
 webcore_built_nosources := DerivedSources/DocTypeStrings.cpp \
 	DerivedSources/tokenizer.cpp DerivedSources/ColorData.c \
 	$(patsubst %.idl,DerivedSources/JS%.h,$(notdir \
@@ -8062,19 +8063,19 @@ webcore_dist := WebCore/ChangeLog WebCore/LICENSE-APPLE \
 	WebCore/html/HTMLEntityNames.gperf \
 	WebCore/html/HTMLAttributeNames.in WebCore/xml/xmlattrs.in \
 	WebCore/xml/XPathGrammar.y
-webcoregtk_cppflags := $(am__append_7) -DWTF_USE_SOUP=1 \
+webcoregtk_cppflags := $(am__append_8) -DWTF_USE_SOUP=1 \
 	-DENABLE_PLUGIN_PACKAGE_SIMPLE_HASH=1 \
 	-I$(srcdir)/WebCore/accessibility/gtk \
 	-I$(srcdir)/WebCore/loader/gtk -I$(srcdir)/WebCore/page/gtk \
 	-I$(srcdir)/WebCore/platform/graphics/cairo \
 	-I$(srcdir)/WebCore/platform/graphics/gtk \
 	-I$(srcdir)/WebCore/platform/gtk \
-	-I$(srcdir)/WebCore/platform/network/soup $(am__append_21) \
-	$(am__append_23)
+	-I$(srcdir)/WebCore/platform/network/soup $(am__append_22) \
+	$(am__append_24)
 webcoregtk_sources := WebCore/bindings/js/ScriptControllerGtk.cpp \
 	WebCore/plugins/gtk/PluginDataGtk.cpp \
 	WebCore/plugins/gtk/PluginPackageGtk.cpp \
-	WebCore/plugins/gtk/PluginViewGtk.cpp $(am__append_17) \
+	WebCore/plugins/gtk/PluginViewGtk.cpp $(am__append_18) \
 	WebCore/accessibility/gtk/AXObjectCacheAtk.cpp \
 	WebCore/accessibility/gtk/AccessibilityObjectAtk.cpp \
 	WebCore/accessibility/gtk/AccessibilityObjectWrapperAtk.cpp \
@@ -8172,8 +8173,8 @@ webcoregtk_sources := WebCore/bindings/js/ScriptControllerGtk.cpp \
 	WebCore/platform/network/soup/ResourceRequest.h \
 	WebCore/platform/network/soup/ResourceResponse.h \
 	WebCore/platform/text/gtk/TextBreakIteratorInternalICUGtk.cpp \
-	$(am__append_22) $(am__append_24) $(am__append_58) \
-	$(am__append_86) $(am__append_116)
+	$(am__append_23) $(am__append_25) $(am__append_59) \
+	$(am__append_87) $(am__append_117)
 webkitgtk_h_api := $(srcdir)/WebKit/gtk/webkit/webkit.h \
 	$(srcdir)/WebKit/gtk/webkit/webkitdefines.h \
 	$(srcdir)/WebKit/gtk/webkit/webkitdownload.h \
@@ -8257,15 +8258,15 @@ webkitgtk_cleanfiles := $(top_builddir)/WebKit/gtk/docs/version.xml \
 #
 # Extra checks and flags
 global_cppflags := -DBUILDING_CAIRO__=1 -DBUILDING_GTK__=1 \
-	-DWTF_CHANGES $(am__append_2) $(am__append_3) $(am__append_4) \
-	$(am__append_5) $(am__append_8) $(am__append_41)
+	-DWTF_CHANGES $(am__append_3) $(am__append_4) $(am__append_5) \
+	$(am__append_6) $(am__append_9) $(am__append_42)
 
 # Default compiler flags
 global_cflags := -Wall -W -Wcast-align -Wchar-subscripts -Wreturn-type \
 	-Wformat -Wformat-security -Wno-format-y2k -Wundef \
 	-Wmissing-format-attribute -Wpointer-arith -Wwrite-strings \
 	-Wno-unused-parameter -Wno-parentheses -fno-exceptions \
-	$(am__append_6)
+	$(am__append_7)
 global_cxxflags := $(global_cflags) $(SYMBOL_VISIBILITY_INLINES) \
 	-fno-rtti
 
@@ -8286,7 +8287,7 @@ corekit_cflags := -fno-strict-aliasing $(COVERAGE_CFLAGS) \
 # before JavaScriptCore in the include path.
 corekit_cppflags := $(global_cppflags) $(webcore_cppflags) \
 	$(webcoregtk_cppflags) $(javascriptcore_cppflags) \
-	$(am__append_1)
+	$(am__append_1) $(am__append_2)
 
 # -no-undefined required for building DLLs on Windows
 # It breaks the build on other platforms, so we use it conditionally
@@ -8300,7 +8301,7 @@ lib_LTLIBRARIES = \
 
 # Convenience libraries
 noinst_LTLIBRARIES = libJavaScriptCore.la libWebCore.la \
-	$(am__append_118)
+	$(am__append_119)
 nodist_EXTRA_libJavaScriptCore_la_SOURCES = \
 	$(javascriptcore_built_nosources)
 
@@ -8477,7 +8478,7 @@ CLEANFILES := $(stamp_files) $(BUILT_SOURCES) \
 	JavaScriptCore/runtime/StringPrototype.lut.h \
 	JavaScriptCore/pcre/chartables.c Programs/jsc Programs/minidom \
 	libWebCoreJS.la Programs/GtkLauncher Programs/DumpRenderTree \
-	$(am__append_120) $(webkitgtk_cleanfiles)
+	$(am__append_121) $(webkitgtk_cleanfiles)
 
 # minidom
 Programs_minidom_SOURCES = \
@@ -8534,22 +8535,22 @@ WebCore := $(srcdir)/WebCore
 # ----
 # HTML Data Grid - enable always.
 # ----
-FEATURE_DEFINES_JAVASCRIPT := LANGUAGE_JAVASCRIPT=1 $(am__append_19) \
-	ENABLE_DATAGRID=1 $(am__append_25) $(am__append_27) \
-	$(am__append_33) $(am__append_38) $(am__append_42) \
-	$(am__append_44) $(am__append_52) $(am__append_62) \
-	$(am__append_67) $(am__append_70) $(am__append_73) \
-	$(am__append_76) $(am__append_80) $(am__append_84) \
-	$(am__append_87) $(am__append_90) $(am__append_97) \
-	$(am__append_100) $(am__append_103) $(am__append_106) \
-	$(am__append_109) $(am__append_112)
-SVG_FEATURES := $(am__append_83) $(am__append_98) $(am__append_101) \
-	$(am__append_104) $(am__append_107) $(am__append_110)
-HTML_FEATURES := $(am__append_39) $(am__append_53) $(am__append_60)
+FEATURE_DEFINES_JAVASCRIPT := LANGUAGE_JAVASCRIPT=1 $(am__append_20) \
+	ENABLE_DATAGRID=1 $(am__append_26) $(am__append_28) \
+	$(am__append_34) $(am__append_39) $(am__append_43) \
+	$(am__append_45) $(am__append_53) $(am__append_63) \
+	$(am__append_68) $(am__append_71) $(am__append_74) \
+	$(am__append_77) $(am__append_81) $(am__append_85) \
+	$(am__append_88) $(am__append_91) $(am__append_98) \
+	$(am__append_101) $(am__append_104) $(am__append_107) \
+	$(am__append_110) $(am__append_113)
+SVG_FEATURES := $(am__append_84) $(am__append_99) $(am__append_102) \
+	$(am__append_105) $(am__append_108) $(am__append_111)
+HTML_FEATURES := $(am__append_40) $(am__append_54) $(am__append_61)
 WEBCORE_CSS_PROPERTY_NAMES := $(WebCore)/css/CSSPropertyNames.in \
-	$(am__append_32) $(am__append_91)
+	$(am__append_33) $(am__append_92)
 WEBCORE_CSS_VALUE_KEYWORDS := $(WebCore)/css/CSSValueKeywords.in \
-	$(am__append_92)
+	$(am__append_93)
 
 # user agent style sheets
 USER_AGENT_STYLE_SHEETS = \
@@ -8663,7 +8664,7 @@ dumprendertree_cppflags := -I$(srcdir)/WebKitTools/DumpRenderTree \
 	-I$(srcdir)/WebKitTools/DumpRenderTree/gtk \
 	-I$(srcdir)/WebKit/gtk -I$(top_builddir)/WebKit/gtk \
 	$(global_cppflags) $(javascriptcore_cppflags) \
-	$(am__append_119)
+	$(am__append_120)
 Programs_DumpRenderTree_CPPFLAGS = $(dumprendertree_cppflags)
 Programs_DumpRenderTree_SOURCES = \
 	WebKitTools/DumpRenderTree/DumpRenderTree.h \
diff --git a/WebCore/bridge/npapi.h b/WebCore/bridge/npapi.h
index 07121b0..b416f67 100644
--- a/WebCore/bridge/npapi.h
+++ b/WebCore/bridge/npapi.h
@@ -78,7 +78,7 @@
 #   endif
 #endif  /* __SYMBIAN32__ */
 
-#if defined(__APPLE_CC__) && !defined(__MACOS_CLASSIC__) && !defined(XP_UNIX)
+#if defined(__APPLE_CC__) && !defined(__MACOS_CLASSIC__) && !defined(XP_UNIX) && !defined(XP_MACOSX)
 #   define XP_MACOSX
 #endif
 
diff --git a/WebCore/plugins/gtk/PluginViewGtk.cpp b/WebCore/plugins/gtk/PluginViewGtk.cpp
index d843401..06984b7 100644
--- a/WebCore/plugins/gtk/PluginViewGtk.cpp
+++ b/WebCore/plugins/gtk/PluginViewGtk.cpp
@@ -446,7 +446,9 @@ bool PluginView::platformStart()
         PluginView::setCurrentPluginView(this);
         JSC::JSLock::DropAllLocks dropAllLocks(JSC::SilenceAssertionsOnly);
         setCallingPlugin(true);
+#ifndef XP_MACOSX
         m_plugin->pluginFuncs()->getvalue(m_instance, NPPVpluginNeedsXEmbed, &m_needsXEmbed);
+#endif
         setCallingPlugin(false);
         PluginView::setCurrentPluginView(0);
     }
diff --git a/WebCore/plugins/pluginview.h b/WebCore/plugins/pluginview.h
index 54c1065..1aa48cc 100644
--- a/WebCore/plugins/pluginview.h
+++ b/WebCore/plugins/pluginview.h
@@ -303,12 +303,12 @@ private:
 #if defined(XP_UNIX) || defined(Q_WS_X11)
         void setNPWindowIfNeeded();
 #elif defined(XP_MACOSX)
+        void setNPWindowIfNeeded();
         NP_CGContext m_npCgContext;
         OwnPtr<Timer<PluginView> > m_nullEventTimer;
         NPDrawingModel m_drawingModel;
         NPEventModel m_eventModel;
 
-        void setNPWindowIfNeeded();
         void nullEventTimerFired(Timer<PluginView>*);
         Point globalMousePosForPlugin() const;
 #endif
Index: WebCore/icu/unicode/usearch.h
===================================================================
--- WebCore/icu/unicode/usearch.h	(revision 55240)
+++ WebCore/icu/unicode/usearch.h	(working copy)
@@ -0,0 +1,643 @@
+/*
+**********************************************************************
+*   Copyright (C) 2001-2005 IBM and others. All rights reserved.
+**********************************************************************
+*   Date        Name        Description
+*  06/28/2001   synwee      Creation.
+**********************************************************************
+*/
+#ifndef USEARCH_H
+#define USEARCH_H
+
+#include "unicode/utypes.h"
+
+#if !UCONFIG_NO_COLLATION
+
+#include "unicode/ucol.h"
+#include "unicode/ucoleitr.h"
+#include "unicode/ubrk.h"
+
+/**
+ * \file
+ * \brief C API: StringSearch
+ *
+ * C Apis for an engine that provides language-sensitive text searching based 
+ * on the comparison rules defined in a <tt>UCollator</tt> data struct,
+ * see <tt>ucol.h</tt>. This ensures that language eccentricity can be 
+ * handled, e.g. for the German collator, characters &szlig; and SS will be matched 
+ * if case is chosen to be ignored. 
+ * See the <a href="http://dev.icu-project.org/cgi-bin/viewcvs.cgi/~checkout~/icuhtml/design/collation/ICU_collation_design.htm">
+ * "ICU Collation Design Document"</a> for more information.
+ * <p> 
+ * The algorithm implemented is a modified form of the Boyer Moore's search.
+ * For more information  see 
+ * <a href="http://icu.sourceforge.net/docs/papers/efficient_text_searching_in_java.html">
+ * "Efficient Text Searching in Java"</a>, published in <i>Java Report</i> 
+ * in February, 1999, for further information on the algorithm.
+ * <p>
+ * There are 2 match options for selection:<br>
+ * Let S' be the sub-string of a text string S between the offsets start and 
+ * end <start, end>.
+ * <br>
+ * A pattern string P matches a text string S at the offsets <start, end> 
+ * if
+ * <pre> 
+ * option 1. Some canonical equivalent of P matches some canonical equivalent 
+ *           of S'
+ * option 2. P matches S' and if P starts or ends with a combining mark, 
+ *           there exists no non-ignorable combining mark before or after S' 
+ *           in S respectively. 
+ * </pre>
+ * Option 2. will be the default.
+ * <p>
+ * This search has APIs similar to that of other text iteration mechanisms 
+ * such as the break iterators in <tt>ubrk.h</tt>. Using these 
+ * APIs, it is easy to scan through text looking for all occurances of 
+ * a given pattern. This search iterator allows changing of direction by 
+ * calling a <tt>reset</tt> followed by a <tt>next</tt> or <tt>previous</tt>. 
+ * Though a direction change can occur without calling <tt>reset</tt> first,  
+ * this operation comes with some speed penalty.
+ * Generally, match results in the forward direction will match the result 
+ * matches in the backwards direction in the reverse order
+ * <p>
+ * <tt>usearch.h</tt> provides APIs to specify the starting position 
+ * within the text string to be searched, e.g. <tt>usearch_setOffset</tt>,
+ * <tt>usearch_preceding</tt> and <tt>usearch_following</tt>. Since the 
+ * starting position will be set as it is specified, please take note that 
+ * there are some dangerous positions which the search may render incorrect 
+ * results:
+ * <ul>
+ * <li> The midst of a substring that requires normalization.
+ * <li> If the following match is to be found, the position should not be the
+ *      second character which requires to be swapped with the preceding 
+ *      character. Vice versa, if the preceding match is to be found, 
+ *      position to search from should not be the first character which 
+ *      requires to be swapped with the next character. E.g certain Thai and
+ *      Lao characters require swapping.
+ * <li> If a following pattern match is to be found, any position within a 
+ *      contracting sequence except the first will fail. Vice versa if a 
+ *      preceding pattern match is to be found, a invalid starting point 
+ *      would be any character within a contracting sequence except the last.
+ * </ul>
+ * <p>
+ * A breakiterator can be used if only matches at logical breaks are desired.
+ * Using a breakiterator will only give you results that exactly matches the
+ * boundaries given by the breakiterator. For instance the pattern "e" will
+ * not be found in the string "\u00e9" if a character break iterator is used.
+ * <p>
+ * Options are provided to handle overlapping matches. 
+ * E.g. In English, overlapping matches produces the result 0 and 2 
+ * for the pattern "abab" in the text "ababab", where else mutually 
+ * exclusive matches only produce the result of 0.
+ * <p>
+ * Though collator attributes will be taken into consideration while 
+ * performing matches, there are no APIs here for setting and getting the 
+ * attributes. These attributes can be set by getting the collator
+ * from <tt>usearch_getCollator</tt> and using the APIs in <tt>ucol.h</tt>.
+ * Lastly to update String Search to the new collator attributes, 
+ * usearch_reset() has to be called.
+ * <p> 
+ * Restriction: <br>
+ * Currently there are no composite characters that consists of a
+ * character with combining class > 0 before a character with combining 
+ * class == 0. However, if such a character exists in the future, the 
+ * search mechanism does not guarantee the results for option 1.
+ * 
+ * <p>
+ * Example of use:<br>
+ * <pre><code>
+ * char *tgtstr = "The quick brown fox jumped over the lazy fox";
+ * char *patstr = "fox";
+ * UChar target[64];
+ * UChar pattern[16];
+ * UErrorCode status = U_ZERO_ERROR;
+ * u_uastrcpy(target, tgtstr);
+ * u_uastrcpy(pattern, patstr);
+ *
+ * UStringSearch *search = usearch_open(pattern, -1, target, -1, "en_US", 
+ *                                  &status);
+ * if (U_SUCCESS(status)) {
+ *     for (int pos = usearch_first(search, &status); 
+ *                                      pos != USEARCH_DONE; 
+ *                                      pos = usearch_next(search, &status)) {
+ *         printf("Found match at %d pos, length is %d\n", pos, 
+ *                                        usearch_getMatchLength(search));
+ *     }
+ * }
+ * </code></pre>
+ * @stable ICU 2.4
+ */
+
+/**
+* DONE is returned by previous() and next() after all valid matches have 
+* been returned, and by first() and last() if there are no matches at all.
+* @stable ICU 2.4
+*/
+#define USEARCH_DONE -1
+
+/**
+* Data structure for searching
+* @stable ICU 2.4
+*/
+struct UStringSearch;
+/**
+* Data structure for searching
+* @stable ICU 2.4
+*/
+typedef struct UStringSearch UStringSearch;
+
+/**
+* @stable ICU 2.4
+*/
+typedef enum {
+    /** Option for overlapping matches */
+    USEARCH_OVERLAP,
+    /** 
+    Option for canonical matches. option 1 in header documentation.
+    The default value will be USEARCH_OFF
+    */
+    USEARCH_CANONICAL_MATCH,
+    USEARCH_ATTRIBUTE_COUNT
+} USearchAttribute;
+
+/**
+* @stable ICU 2.4
+*/
+typedef enum {
+    /** default value for any USearchAttribute */
+    USEARCH_DEFAULT = -1,
+    /** value for USEARCH_OVERLAP and USEARCH_CANONICAL_MATCH */
+    USEARCH_OFF, 
+    /** value for USEARCH_OVERLAP and USEARCH_CANONICAL_MATCH */
+    USEARCH_ON,
+    USEARCH_ATTRIBUTE_VALUE_COUNT
+} USearchAttributeValue;
+
+/* open and close ------------------------------------------------------ */
+
+/**
+* Creating a search iterator data struct using the argument locale language
+* rule set. A collator will be created in the process, which will be owned by
+* this search and will be deleted in <tt>usearch_close</tt>.
+* @param pattern for matching
+* @param patternlength length of the pattern, -1 for null-termination
+* @param text text string
+* @param textlength length of the text string, -1 for null-termination
+* @param locale name of locale for the rules to be used
+* @param breakiter A BreakIterator that will be used to restrict the points
+*                  at which matches are detected. If a match is found, but 
+*                  the match's start or end index is not a boundary as 
+*                  determined by the <tt>BreakIterator</tt>, the match will 
+*                  be rejected and another will be searched for. 
+*                  If this parameter is <tt>NULL</tt>, no break detection is 
+*                  attempted.
+* @param status for errors if it occurs. If pattern or text is NULL, or if
+*               patternlength or textlength is 0 then an 
+*               U_ILLEGAL_ARGUMENT_ERROR is returned.
+* @return search iterator data structure, or NULL if there is an error.
+* @stable ICU 2.4
+*/
+U_STABLE UStringSearch * U_EXPORT2 usearch_open(const UChar          *pattern, 
+                                              int32_t         patternlength, 
+                                        const UChar          *text, 
+                                              int32_t         textlength,
+                                        const char           *locale,
+                                              UBreakIterator *breakiter,
+                                              UErrorCode     *status);
+
+/**
+* Creating a search iterator data struct using the argument collator language
+* rule set. Note, user retains the ownership of this collator, thus the 
+* responsibility of deletion lies with the user.
+* NOTE: string search cannot be instantiated from a collator that has 
+* collate digits as numbers (CODAN) turned on.
+* @param pattern for matching
+* @param patternlength length of the pattern, -1 for null-termination
+* @param text text string
+* @param textlength length of the text string, -1 for null-termination
+* @param collator used for the language rules
+* @param breakiter A BreakIterator that will be used to restrict the points
+*                  at which matches are detected. If a match is found, but 
+*                  the match's start or end index is not a boundary as 
+*                  determined by the <tt>BreakIterator</tt>, the match will 
+*                  be rejected and another will be searched for. 
+*                  If this parameter is <tt>NULL</tt>, no break detection is 
+*                  attempted.
+* @param status for errors if it occurs. If collator, pattern or text is NULL, 
+*               or if patternlength or textlength is 0 then an 
+*               U_ILLEGAL_ARGUMENT_ERROR is returned.
+* @return search iterator data structure, or NULL if there is an error.
+* @stable ICU 2.4
+*/
+U_STABLE UStringSearch * U_EXPORT2 usearch_openFromCollator(
+                                         const UChar *pattern, 
+                                               int32_t         patternlength,
+                                         const UChar          *text, 
+                                               int32_t         textlength,
+                                         const UCollator      *collator,
+                                               UBreakIterator *breakiter,
+                                               UErrorCode     *status);
+
+/**
+* Destroying and cleaning up the search iterator data struct.
+* If a collator is created in <tt>usearch_open</tt>, it will be destroyed here.
+* @param searchiter data struct to clean up
+* @stable ICU 2.4
+*/
+U_STABLE void U_EXPORT2 usearch_close(UStringSearch *searchiter);
+
+/* get and set methods -------------------------------------------------- */
+
+/**
+* Sets the current position in the text string which the next search will 
+* start from. Clears previous states. 
+* This method takes the argument index and sets the position in the text 
+* string accordingly without checking if the index is pointing to a 
+* valid starting point to begin searching. 
+* Search positions that may render incorrect results are highlighted in the
+* header comments
+* @param strsrch search iterator data struct
+* @param position position to start next search from. If position is less
+*          than or greater than the text range for searching, 
+*          an U_INDEX_OUTOFBOUNDS_ERROR will be returned
+* @param status error status if any.
+* @stable ICU 2.4
+*/
+U_STABLE void U_EXPORT2 usearch_setOffset(UStringSearch *strsrch, 
+                                        int32_t    position,
+                                        UErrorCode    *status);
+
+/**
+* Return the current index in the string text being searched.
+* If the iteration has gone past the end of the text (or past the beginning 
+* for a backwards search), <tt>USEARCH_DONE</tt> is returned.
+* @param strsrch search iterator data struct
+* @see #USEARCH_DONE
+* @stable ICU 2.4
+*/
+U_STABLE int32_t U_EXPORT2 usearch_getOffset(const UStringSearch *strsrch);
+    
+/**
+* Sets the text searching attributes located in the enum USearchAttribute
+* with values from the enum USearchAttributeValue.
+* <tt>USEARCH_DEFAULT</tt> can be used for all attributes for resetting.
+* @param strsrch search iterator data struct
+* @param attribute text attribute to be set
+* @param value text attribute value
+* @param status for errors if it occurs
+* @see #usearch_getAttribute
+* @stable ICU 2.4
+*/
+U_STABLE void U_EXPORT2 usearch_setAttribute(UStringSearch         *strsrch, 
+                                           USearchAttribute       attribute,
+                                           USearchAttributeValue  value,
+                                           UErrorCode            *status);
+
+/**    
+* Gets the text searching attributes.
+* @param strsrch search iterator data struct
+* @param attribute text attribute to be retrieve
+* @return text attribute value
+* @see #usearch_setAttribute
+* @stable ICU 2.4
+*/
+U_STABLE USearchAttributeValue U_EXPORT2 usearch_getAttribute(
+                                         const UStringSearch    *strsrch,
+                                               USearchAttribute  attribute);
+
+/**
+* Returns the index to the match in the text string that was searched.
+* This call returns a valid result only after a successful call to 
+* <tt>usearch_first</tt>, <tt>usearch_next</tt>, <tt>usearch_previous</tt>, 
+* or <tt>usearch_last</tt>.
+* Just after construction, or after a searching method returns 
+* <tt>USEARCH_DONE</tt>, this method will return <tt>USEARCH_DONE</tt>.
+* <p>
+* Use <tt>usearch_getMatchedLength</tt> to get the matched string length.
+* @param strsrch search iterator data struct
+* @return index to a substring within the text string that is being 
+*         searched.
+* @see #usearch_first
+* @see #usearch_next
+* @see #usearch_previous
+* @see #usearch_last
+* @see #USEARCH_DONE
+* @stable ICU 2.4
+*/
+U_STABLE int32_t U_EXPORT2 usearch_getMatchedStart(
+                                               const UStringSearch *strsrch);
+    
+/**
+* Returns the length of text in the string which matches the search pattern. 
+* This call returns a valid result only after a successful call to 
+* <tt>usearch_first</tt>, <tt>usearch_next</tt>, <tt>usearch_previous</tt>, 
+* or <tt>usearch_last</tt>.
+* Just after construction, or after a searching method returns 
+* <tt>USEARCH_DONE</tt>, this method will return 0.
+* @param strsrch search iterator data struct
+* @return The length of the match in the string text, or 0 if there is no 
+*         match currently.
+* @see #usearch_first
+* @see #usearch_next
+* @see #usearch_previous
+* @see #usearch_last
+* @see #USEARCH_DONE
+* @stable ICU 2.4
+*/
+U_STABLE int32_t U_EXPORT2 usearch_getMatchedLength(
+                                               const UStringSearch *strsrch);
+
+/**
+* Returns the text that was matched by the most recent call to 
+* <tt>usearch_first</tt>, <tt>usearch_next</tt>, <tt>usearch_previous</tt>, 
+* or <tt>usearch_last</tt>.
+* If the iterator is not pointing at a valid match (e.g. just after 
+* construction or after <tt>USEARCH_DONE</tt> has been returned, returns
+* an empty string. If result is not large enough to store the matched text,
+* result will be filled with the partial text and an U_BUFFER_OVERFLOW_ERROR 
+* will be returned in status. result will be null-terminated whenever 
+* possible. If the buffer fits the matched text exactly, a null-termination 
+* is not possible, then a U_STRING_NOT_TERMINATED_ERROR set in status.
+* Pre-flighting can be either done with length = 0 or the API 
+* <tt>usearch_getMatchLength</tt>.
+* @param strsrch search iterator data struct
+* @param result UChar buffer to store the matched string
+* @param resultCapacity length of the result buffer
+* @param status error returned if result is not large enough
+* @return exact length of the matched text, not counting the null-termination
+* @see #usearch_first
+* @see #usearch_next
+* @see #usearch_previous
+* @see #usearch_last
+* @see #USEARCH_DONE
+* @stable ICU 2.4
+*/
+U_STABLE int32_t U_EXPORT2 usearch_getMatchedText(const UStringSearch *strsrch, 
+                                            UChar         *result, 
+                                            int32_t        resultCapacity, 
+                                            UErrorCode    *status);
+
+#if !UCONFIG_NO_BREAK_ITERATION
+
+/**
+* Set the BreakIterator that will be used to restrict the points at which 
+* matches are detected.
+* @param strsrch search iterator data struct
+* @param breakiter A BreakIterator that will be used to restrict the points
+*                  at which matches are detected. If a match is found, but 
+*                  the match's start or end index is not a boundary as 
+*                  determined by the <tt>BreakIterator</tt>, the match will 
+*                  be rejected and another will be searched for. 
+*                  If this parameter is <tt>NULL</tt>, no break detection is 
+*                  attempted.
+* @param status for errors if it occurs
+* @see #usearch_getBreakIterator
+* @stable ICU 2.4
+*/
+U_STABLE void U_EXPORT2 usearch_setBreakIterator(UStringSearch  *strsrch, 
+                                               UBreakIterator *breakiter,
+                                               UErrorCode     *status);
+
+/**
+* Returns the BreakIterator that is used to restrict the points at which 
+* matches are detected. This will be the same object that was passed to the 
+* constructor or to <tt>usearch_setBreakIterator</tt>. Note that 
+* <tt>NULL</tt> 
+* is a legal value; it means that break detection should not be attempted.
+* @param strsrch search iterator data struct
+* @return break iterator used
+* @see #usearch_setBreakIterator
+* @stable ICU 2.4
+*/
+U_STABLE const UBreakIterator * U_EXPORT2 usearch_getBreakIterator(
+                                              const UStringSearch *strsrch);
+    
+#endif
+    
+/**
+* Set the string text to be searched. Text iteration will hence begin at the 
+* start of the text string. This method is useful if you want to re-use an 
+* iterator to search for the same pattern within a different body of text.
+* @param strsrch search iterator data struct
+* @param text new string to look for match
+* @param textlength length of the new string, -1 for null-termination
+* @param status for errors if it occurs. If text is NULL, or textlength is 0 
+*               then an U_ILLEGAL_ARGUMENT_ERROR is returned with no change
+*               done to strsrch.
+* @see #usearch_getText
+* @stable ICU 2.4
+*/
+U_STABLE void U_EXPORT2 usearch_setText(      UStringSearch *strsrch, 
+                                      const UChar         *text,
+                                            int32_t        textlength,
+                                            UErrorCode    *status);
+
+/**
+* Return the string text to be searched.
+* @param strsrch search iterator data struct
+* @param length returned string text length
+* @return string text 
+* @see #usearch_setText
+* @stable ICU 2.4
+*/
+U_STABLE const UChar * U_EXPORT2 usearch_getText(const UStringSearch *strsrch, 
+                                               int32_t       *length);
+
+/**
+* Gets the collator used for the language rules. 
+* <p>
+* Deleting the returned <tt>UCollator</tt> before calling 
+* <tt>usearch_close</tt> would cause the string search to fail.
+* <tt>usearch_close</tt> will delete the collator if this search owns it.
+* @param strsrch search iterator data struct
+* @return collator
+* @stable ICU 2.4
+*/
+U_STABLE UCollator * U_EXPORT2 usearch_getCollator(
+                                               const UStringSearch *strsrch);
+
+/**
+* Sets the collator used for the language rules. User retains the ownership 
+* of this collator, thus the responsibility of deletion lies with the user.
+* This method causes internal data such as Boyer-Moore shift tables to  
+* be recalculated, but the iterator's position is unchanged.
+* @param strsrch search iterator data struct
+* @param collator to be used
+* @param status for errors if it occurs
+* @stable ICU 2.4
+*/
+U_STABLE void U_EXPORT2 usearch_setCollator(      UStringSearch *strsrch, 
+                                          const UCollator     *collator,
+                                                UErrorCode    *status);
+
+/**
+* Sets the pattern used for matching.
+* Internal data like the Boyer Moore table will be recalculated, but the 
+* iterator's position is unchanged.
+* @param strsrch search iterator data struct
+* @param pattern string
+* @param patternlength pattern length, -1 for null-terminated string
+* @param status for errors if it occurs. If text is NULL, or textlength is 0 
+*               then an U_ILLEGAL_ARGUMENT_ERROR is returned with no change
+*               done to strsrch.
+* @stable ICU 2.4
+*/
+U_STABLE void U_EXPORT2 usearch_setPattern(      UStringSearch *strsrch, 
+                                         const UChar         *pattern,
+                                               int32_t        patternlength,
+                                               UErrorCode    *status);
+
+/**
+* Gets the search pattern
+* @param strsrch search iterator data struct
+* @param length return length of the pattern, -1 indicates that the pattern 
+*               is null-terminated
+* @return pattern string
+* @stable ICU 2.4
+*/
+U_STABLE const UChar * U_EXPORT2 usearch_getPattern(
+                                               const UStringSearch *strsrch, 
+                                                     int32_t       *length);
+
+/* methods ------------------------------------------------------------- */
+
+/**
+* Returns the first index at which the string text matches the search 
+* pattern.  
+* The iterator is adjusted so that its current index (as returned by 
+* <tt>usearch_getOffset</tt>) is the match position if one was found.
+* If a match is not found, <tt>USEARCH_DONE</tt> will be returned and
+* the iterator will be adjusted to the index <tt>USEARCH_DONE</tt>.
+* @param strsrch search iterator data struct
+* @param status for errors if it occurs
+* @return The character index of the first match, or 
+* <tt>USEARCH_DONE</tt> if there are no matches.
+* @see #usearch_getOffset
+* @see #USEARCH_DONE
+* @stable ICU 2.4
+*/
+U_STABLE int32_t U_EXPORT2 usearch_first(UStringSearch *strsrch, 
+                                           UErrorCode    *status);
+
+/**
+* Returns the first index greater than <tt>position</tt> at which the string 
+* text 
+* matches the search pattern. The iterator is adjusted so that its current 
+* index (as returned by <tt>usearch_getOffset</tt>) is the match position if 
+* one was found.
+* If a match is not found, <tt>USEARCH_DONE</tt> will be returned and
+* the iterator will be adjusted to the index <tt>USEARCH_DONE</tt>
+* <p>
+* Search positions that may render incorrect results are highlighted in the
+* header comments. If position is less than or greater than the text range 
+* for searching, an U_INDEX_OUTOFBOUNDS_ERROR will be returned
+* @param strsrch search iterator data struct
+* @param position to start the search at
+* @param status for errors if it occurs
+* @return The character index of the first match following <tt>pos</tt>,
+*         or <tt>USEARCH_DONE</tt> if there are no matches.
+* @see #usearch_getOffset
+* @see #USEARCH_DONE
+* @stable ICU 2.4
+*/
+U_STABLE int32_t U_EXPORT2 usearch_following(UStringSearch *strsrch, 
+                                               int32_t    position, 
+                                               UErrorCode    *status);
+    
+/**
+* Returns the last index in the target text at which it matches the search 
+* pattern. The iterator is adjusted so that its current 
+* index (as returned by <tt>usearch_getOffset</tt>) is the match position if 
+* one was found.
+* If a match is not found, <tt>USEARCH_DONE</tt> will be returned and
+* the iterator will be adjusted to the index <tt>USEARCH_DONE</tt>.
+* @param strsrch search iterator data struct
+* @param status for errors if it occurs
+* @return The index of the first match, or <tt>USEARCH_DONE</tt> if there 
+*         are no matches.
+* @see #usearch_getOffset
+* @see #USEARCH_DONE
+* @stable ICU 2.4
+*/
+U_STABLE int32_t U_EXPORT2 usearch_last(UStringSearch *strsrch, 
+                                          UErrorCode    *status);
+
+/**
+* Returns the first index less than <tt>position</tt> at which the string text 
+* matches the search pattern. The iterator is adjusted so that its current 
+* index (as returned by <tt>usearch_getOffset</tt>) is the match position if 
+* one was found.
+* If a match is not found, <tt>USEARCH_DONE</tt> will be returned and
+* the iterator will be adjusted to the index <tt>USEARCH_DONE</tt>
+* <p>
+* Search positions that may render incorrect results are highlighted in the
+* header comments. If position is less than or greater than the text range 
+* for searching, an U_INDEX_OUTOFBOUNDS_ERROR will be returned
+* @param strsrch search iterator data struct
+* @param position index position the search is to begin at
+* @param status for errors if it occurs
+* @return The character index of the first match preceding <tt>pos</tt>,
+*         or <tt>USEARCH_DONE</tt> if there are no matches.
+* @see #usearch_getOffset
+* @see #USEARCH_DONE
+* @stable ICU 2.4
+*/
+U_STABLE int32_t U_EXPORT2 usearch_preceding(UStringSearch *strsrch, 
+                                               int32_t    position, 
+                                               UErrorCode    *status);
+    
+/**
+* Returns the index of the next point at which the string text matches the
+* search pattern, starting from the current position.
+* The iterator is adjusted so that its current 
+* index (as returned by <tt>usearch_getOffset</tt>) is the match position if 
+* one was found.
+* If a match is not found, <tt>USEARCH_DONE</tt> will be returned and
+* the iterator will be adjusted to the index <tt>USEARCH_DONE</tt>
+* @param strsrch search iterator data struct
+* @param status for errors if it occurs
+* @return The index of the next match after the current position, or 
+*         <tt>USEARCH_DONE</tt> if there are no more matches.
+* @see #usearch_first
+* @see #usearch_getOffset
+* @see #USEARCH_DONE
+* @stable ICU 2.4
+*/
+U_STABLE int32_t U_EXPORT2 usearch_next(UStringSearch *strsrch, 
+                                          UErrorCode    *status);
+
+/**
+* Returns the index of the previous point at which the string text matches
+* the search pattern, starting at the current position.
+* The iterator is adjusted so that its current 
+* index (as returned by <tt>usearch_getOffset</tt>) is the match position if 
+* one was found.
+* If a match is not found, <tt>USEARCH_DONE</tt> will be returned and
+* the iterator will be adjusted to the index <tt>USEARCH_DONE</tt>
+* @param strsrch search iterator data struct
+* @param status for errors if it occurs
+* @return The index of the previous match before the current position,
+*         or <tt>USEARCH_DONE</tt> if there are no more matches.
+* @see #usearch_last
+* @see #usearch_getOffset
+* @see #USEARCH_DONE
+* @stable ICU 2.4
+*/
+U_STABLE int32_t U_EXPORT2 usearch_previous(UStringSearch *strsrch, 
+                                              UErrorCode    *status);
+    
+/** 
+* Reset the iteration.
+* Search will begin at the start of the text string if a forward iteration 
+* is initiated before a backwards iteration. Otherwise if a backwards 
+* iteration is initiated before a forwards iteration, the search will begin
+* at the end of the text string.
+* @param strsrch search iterator data struct
+* @see #usearch_first
+* @stable ICU 2.4
+*/
+U_STABLE void U_EXPORT2 usearch_reset(UStringSearch *strsrch);
+
+#endif /* #if !UCONFIG_NO_COLLATION */
+
+#endif
Index: WebCore/icu/unicode/ucoleitr.h
===================================================================
+-- WebCore/icu/unicode/ucoleitr.h	(revision 55240)
+++ WebCore/icu/unicode/ucoleitr.h	(working copy)
@@ -0,0 +1,268 @@
+/*
+*******************************************************************************
+*   Copyright (C) 2001-2004, International Business Machines
+*   Corporation and others.  All Rights Reserved.
+*******************************************************************************
+*
+* File ucoleitr.cpp
+*
+* Modification History:
+*
+* Date        Name        Description
+* 02/15/2001  synwee      Modified all methods to process its own function 
+*                         instead of calling the equivalent c++ api (coleitr.h)
+*******************************************************************************/
+
+#ifndef UCOLEITR_H
+#define UCOLEITR_H
+
+#include "unicode/utypes.h"
+
+#if !UCONFIG_NO_COLLATION
+
+/**  
+ * This indicates an error has occured during processing or if no more CEs is 
+ * to be returned.
+ * @stable ICU 2.0
+ */
+#define UCOL_NULLORDER        ((int32_t)0xFFFFFFFF)
+
+#include "unicode/ucol.h"
+
+/** 
+ * The UCollationElements struct.
+ * For usage in C programs.
+ * @stable ICU 2.0
+ */
+typedef struct UCollationElements UCollationElements;
+
+/**
+ * \file
+ * \brief C API: UCollationElements
+ *
+ * The UCollationElements API is used as an iterator to walk through each 
+ * character of an international string. Use the iterator to return the
+ * ordering priority of the positioned character. The ordering priority of a 
+ * character, which we refer to as a key, defines how a character is collated 
+ * in the given collation object.
+ * For example, consider the following in Spanish:
+ * <pre>
+ * .       "ca" -> the first key is key('c') and second key is key('a').
+ * .       "cha" -> the first key is key('ch') and second key is key('a').
+ * </pre>
+ * And in German,
+ * <pre>
+ * .       "<ae ligature>b"-> the first key is key('a'), the second key is key('e'), and
+ * .       the third key is key('b').
+ * </pre>
+ * <p>Example of the iterator usage: (without error checking)
+ * <pre>
+ * .  void CollationElementIterator_Example()
+ * .  {
+ * .      UChar *s;
+ * .      t_int32 order, primaryOrder;
+ * .      UCollationElements *c;
+ * .      UCollatorOld *coll;
+ * .      UErrorCode success = U_ZERO_ERROR;
+ * .      s=(UChar*)malloc(sizeof(UChar) * (strlen("This is a test")+1) );
+ * .      u_uastrcpy(s, "This is a test");
+ * .      coll = ucol_open(NULL, &success);
+ * .      c = ucol_openElements(coll, str, u_strlen(str), &status);
+ * .      order = ucol_next(c, &success);
+ * .      ucol_reset(c);
+ * .      order = ucol_prev(c, &success);
+ * .      free(s);
+ * .      ucol_close(coll);
+ * .      ucol_closeElements(c);
+ * .  }
+ * </pre>
+ * <p>
+ * ucol_next() returns the collation order of the next.
+ * ucol_prev() returns the collation order of the previous character.
+ * The Collation Element Iterator moves only in one direction between calls to
+ * ucol_reset. That is, ucol_next() and ucol_prev can not be inter-used. 
+ * Whenever ucol_prev is to be called after ucol_next() or vice versa, 
+ * ucol_reset has to be called first to reset the status, shifting pointers to 
+ * either the end or the start of the string. Hence at the next call of 
+ * ucol_prev or ucol_next, the first or last collation order will be returned. 
+ * If a change of direction is done without a ucol_reset, the result is 
+ * undefined.
+ * The result of a forward iterate (ucol_next) and reversed result of the  
+ * backward iterate (ucol_prev) on the same string are equivalent, if 
+ * collation orders with the value UCOL_IGNORABLE are ignored.
+ * Character based on the comparison level of the collator.  A collation order 
+ * consists of primary order, secondary order and tertiary order.  The data 
+ * type of the collation order is <strong>t_int32</strong>. 
+ *
+ * @see UCollator
+ */
+
+/**
+ * Open the collation elements for a string.
+ *
+ * @param coll The collator containing the desired collation rules.
+ * @param text The text to iterate over.
+ * @param textLength The number of characters in text, or -1 if null-terminated
+ * @param status A pointer to an UErrorCode to receive any errors.
+ * @return a struct containing collation element information
+ * @stable ICU 2.0
+ */
+U_STABLE UCollationElements* U_EXPORT2 
+ucol_openElements(const UCollator  *coll,
+                  const UChar      *text,
+                        int32_t    textLength,
+                        UErrorCode *status);
+
+/**
+ * get a hash code for a key... Not very useful!
+ * @param key    the given key.
+ * @param length the size of the key array.
+ * @return       the hash code.
+ * @stable ICU 2.0
+ */
+U_STABLE int32_t U_EXPORT2 
+ucol_keyHashCode(const uint8_t* key, int32_t length);
+
+/**
+ * Close a UCollationElements.
+ * Once closed, a UCollationElements may no longer be used.
+ * @param elems The UCollationElements to close.
+ * @stable ICU 2.0
+ */
+U_STABLE void U_EXPORT2 
+ucol_closeElements(UCollationElements *elems);
+
+/**
+ * Reset the collation elements to their initial state.
+ * This will move the 'cursor' to the beginning of the text.
+ * Property settings for collation will be reset to the current status.
+ * @param elems The UCollationElements to reset.
+ * @see ucol_next
+ * @see ucol_previous
+ * @stable ICU 2.0
+ */
+U_STABLE void U_EXPORT2 
+ucol_reset(UCollationElements *elems);
+
+/**
+ * Get the ordering priority of the next collation element in the text.
+ * A single character may contain more than one collation element.
+ * @param elems The UCollationElements containing the text.
+ * @param status A pointer to an UErrorCode to receive any errors.
+ * @return The next collation elements ordering, otherwise returns NULLORDER 
+ *         if an error has occured or if the end of string has been reached
+ * @stable ICU 2.0
+ */
+U_STABLE int32_t U_EXPORT2 
+ucol_next(UCollationElements *elems, UErrorCode *status);
+
+/**
+ * Get the ordering priority of the previous collation element in the text.
+ * A single character may contain more than one collation element.
+ * Note that internally a stack is used to store buffered collation elements. 
+ * It is very rare that the stack will overflow, however if such a case is 
+ * encountered, the problem can be solved by increasing the size 
+ * UCOL_EXPAND_CE_BUFFER_SIZE in ucol_imp.h.
+ * @param elems The UCollationElements containing the text.
+ * @param status A pointer to an UErrorCode to receive any errors. Noteably 
+ *               a U_BUFFER_OVERFLOW_ERROR is returned if the internal stack
+ *               buffer has been exhausted.
+ * @return The previous collation elements ordering, otherwise returns 
+ *         NULLORDER if an error has occured or if the start of string has 
+ *         been reached.
+ * @stable ICU 2.0
+ */
+U_STABLE int32_t U_EXPORT2 
+ucol_previous(UCollationElements *elems, UErrorCode *status);
+
+/**
+ * Get the maximum length of any expansion sequences that end with the 
+ * specified comparison order.
+ * This is useful for .... ?
+ * @param elems The UCollationElements containing the text.
+ * @param order A collation order returned by previous or next.
+ * @return maximum size of the expansion sequences ending with the collation 
+ *         element or 1 if collation element does not occur at the end of any 
+ *         expansion sequence
+ * @stable ICU 2.0
+ */
+U_STABLE int32_t U_EXPORT2 
+ucol_getMaxExpansion(const UCollationElements *elems, int32_t order);
+
+/**
+ * Set the text containing the collation elements.
+ * Property settings for collation will remain the same.
+ * In order to reset the iterator to the current collation property settings,
+ * the API reset() has to be called.
+ * @param elems The UCollationElements to set.
+ * @param text The source text containing the collation elements.
+ * @param textLength The length of text, or -1 if null-terminated.
+ * @param status A pointer to an UErrorCode to receive any errors.
+ * @see ucol_getText
+ * @stable ICU 2.0
+ */
+U_STABLE void U_EXPORT2 
+ucol_setText(      UCollationElements *elems, 
+             const UChar              *text,
+                   int32_t            textLength,
+                   UErrorCode         *status);
+
+/**
+ * Get the offset of the current source character.
+ * This is an offset into the text of the character containing the current
+ * collation elements.
+ * @param elems The UCollationElements to query.
+ * @return The offset of the current source character.
+ * @see ucol_setOffset
+ * @stable ICU 2.0
+ */
+U_STABLE int32_t U_EXPORT2 
+ucol_getOffset(const UCollationElements *elems);
+
+/**
+ * Set the offset of the current source character.
+ * This is an offset into the text of the character to be processed.
+ * Property settings for collation will remain the same.
+ * In order to reset the iterator to the current collation property settings,
+ * the API reset() has to be called.
+ * @param elems The UCollationElements to set.
+ * @param offset The desired character offset.
+ * @param status A pointer to an UErrorCode to receive any errors.
+ * @see ucol_getOffset
+ * @stable ICU 2.0
+ */
+U_STABLE void U_EXPORT2 
+ucol_setOffset(UCollationElements *elems,
+               int32_t        offset,
+               UErrorCode         *status);
+
+/**
+* Get the primary order of a collation order.
+* @param order the collation order
+* @return the primary order of a collation order.
+* @stable ICU 2.6
+*/
+U_STABLE int32_t U_EXPORT2
+ucol_primaryOrder (int32_t order); 
+
+/**
+* Get the secondary order of a collation order.
+* @param order the collation order
+* @return the secondary order of a collation order.
+* @stable ICU 2.6
+*/
+U_STABLE int32_t U_EXPORT2
+ucol_secondaryOrder (int32_t order); 
+
+/**
+* Get the tertiary order of a collation order.
+* @param order the collation order
+* @return the tertiary order of a collation order.
+* @stable ICU 2.6
+*/
+U_STABLE int32_t U_EXPORT2
+ucol_tertiaryOrder (int32_t order); 
+
+#endif /* #if !UCONFIG_NO_COLLATION */
+
+#endif
